{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "bf35c37f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MultiDiGraph with 6422 nodes and 669305 edges\n"
     ]
    }
   ],
   "source": [
    "import networkx as nx\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "from networkx.readwrite import json_graph\n",
    "\n",
    "import os, json\n",
    "\n",
    "with open(\"data/regelems_KG.json\") as f:\n",
    "    data = json.load(f)\n",
    "\n",
    "G: nx.MultiDiGraph = json_graph.node_link_graph(data, edges=\"edges\")\n",
    "print(G)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1af9aab6",
   "metadata": {},
   "outputs": [],
   "source": [
    "def node_color(x):\n",
    "    return \"lightblue\" if type(x) == str else \"red\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "929b7b68",
   "metadata": {},
   "outputs": [],
   "source": [
    "# attempt visualization\n",
    "#plt.figure(figsize=(12, 12))\n",
    "#pos = nx.spring_layout(G)\n",
    "#nx.draw(G, pos, with_labels=False, node_size=50, font_size=8, node_color=[node_color(n) for n in G.nodes()])\n",
    "#plt.title(\"NCBI Functional Regulatory Elements Knowledge Graph\")\n",
    "#plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7351f6b3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Top 10 nodes by in-degree centrality:\n",
      "Node: 8161, In-Degree Centrality: 44.1043\n",
      "Node: 7030, In-Degree Centrality: 16.6837\n",
      "Node: 1422, In-Degree Centrality: 1.2032\n",
      "Node: 3378, In-Degree Centrality: 1.0486\n",
      "Node: 9414, In-Degree Centrality: 1.0477\n",
      "Node: 4831, In-Degree Centrality: 0.9868\n",
      "Node: 5984, In-Degree Centrality: 0.9541\n",
      "Node: 7949, In-Degree Centrality: 0.9369\n",
      "Node: 5741, In-Degree Centrality: 0.9047\n",
      "Node: 9021, In-Degree Centrality: 0.7746\n",
      "\n",
      "Top 10 nodes by out-degree centrality:\n",
      "Node: <cls>, Out-Degree Centrality: 0.8478\n",
      "Node: G, Out-Degree Centrality: 0.3898\n",
      "Node: C, Out-Degree Centrality: 0.3755\n",
      "Node: T, Out-Degree Centrality: 0.3398\n",
      "Node: A, Out-Degree Centrality: 0.3269\n",
      "Node: TTTTTT, Out-Degree Centrality: 0.1448\n",
      "Node: AAAAAA, Out-Degree Centrality: 0.1420\n",
      "Node: ATTTTT, Out-Degree Centrality: 0.0941\n",
      "Node: AAAAAT, Out-Degree Centrality: 0.0927\n",
      "Node: TATTTT, Out-Degree Centrality: 0.0911\n",
      "\n",
      "Top 10 nodes by betweenness centrality:\n",
      "Node: 8161, Betweenness Centrality: 0.1504\n",
      "Node: 7030, Betweenness Centrality: 0.1143\n",
      "Node: 9414, Betweenness Centrality: 0.0540\n",
      "Node: 1422, Betweenness Centrality: 0.0466\n",
      "Node: 5984, Betweenness Centrality: 0.0420\n",
      "Node: 9021, Betweenness Centrality: 0.0377\n",
      "Node: 7949, Betweenness Centrality: 0.0273\n",
      "Node: 2453, Betweenness Centrality: 0.0254\n",
      "Node: 9452, Betweenness Centrality: 0.0253\n",
      "Node: 5921, Betweenness Centrality: 0.0253\n"
     ]
    }
   ],
   "source": [
    "# centrality\n",
    "\n",
    "in_degree = nx.in_degree_centrality(G)\n",
    "in_sorted = sorted(in_degree.items(), key=lambda x: x[1], reverse=True)\n",
    "print(\"Top 10 nodes by in-degree centrality:\")\n",
    "for node, centrality in in_sorted[:10]:\n",
    "    print(f\"Node: {node}, In-Degree Centrality: {centrality:.4f}\")\n",
    "\n",
    "out_degree = nx.out_degree_centrality(G)\n",
    "out_sorted = sorted(out_degree.items(), key=lambda x: x[1], reverse=True)\n",
    "print(\"\\nTop 10 nodes by out-degree centrality:\")\n",
    "for node, centrality in out_sorted[:10]:\n",
    "    print(f\"Node: {node}, Out-Degree Centrality: {centrality:.4f}\")\n",
    "\n",
    "NDG = G.to_undirected()\n",
    "betweenness = nx.betweenness_centrality(NDG)\n",
    "betweenness_sorted = sorted(betweenness.items(), key=lambda x: x[1], reverse=True)\n",
    "print(\"\\nTop 10 nodes by betweenness centrality:\")\n",
    "for node, centrality in betweenness_sorted[:10]:\n",
    "    print(f\"Node: {node}, Betweenness Centrality: {centrality:.4f}\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fcc6976e",
   "metadata": {},
   "source": [
    "# Looking at f/8161 and f/7030 as Regulatory Element Indicators\n",
    "\n",
    "Previous enrichment results seem to show f/8161 and f/7030 are strongly associated with regulatory functions, and intervention results suggest that f/8161 may also identify some sort of inhibitory factor which prevents cisplatin from binding transcripts with this feature active. Given this, we should examine the neighborhood around these features and repeat our GSEA to see if we find similar functional enrichment. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "7019a0e1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MultiGraph with 6398 nodes and 667936 edges\n",
      "MultiGraph with 6035 nodes and 647893 edges\n"
     ]
    }
   ],
   "source": [
    "# Pick out the two hop neighborhoods of f/8161 and f/7030\n",
    "neighbors_8161 = set(nx.single_source_shortest_path_length(NDG, 8161, cutoff=2).keys())\n",
    "neighbors_7030 = set(nx.single_source_shortest_path_length(NDG, 7030, cutoff=2).keys())\n",
    "\n",
    "# visualize the neighborhoods\n",
    "subgraph_8161 = NDG.subgraph(neighbors_8161)\n",
    "print(subgraph_8161)\n",
    "subgraph_7030 = NDG.subgraph(neighbors_7030)\n",
    "print(subgraph_7030)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cced9f17",
   "metadata": {},
   "source": [
    "Since the two-hop neighborhood nearly encompassses the entire graph, skip visualization"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "6bb76b1a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Neighborhood Size of f/8161: 4078 / 175 genes\n",
      "GC content of motifs associated with f/8161: 0.49861860389079615\n",
      "Neighborhood Size of f/7030: 3728 / 178 genes\n",
      "GC content of motifs associated with f/7030: 0.48676680972818315\n"
     ]
    }
   ],
   "source": [
    "def feature_geneandmotif_sets(center, edges: list) -> tuple[set, set]:\n",
    "    genes = []\n",
    "    motifs = []\n",
    "    for u, v, annotations in edges:\n",
    "        motif = u if u != center else v\n",
    "        if motif:\n",
    "            motifs.append(motif)        \n",
    "        for key, val in annotations.items():\n",
    "            genes.append(val[\"gene_id\"])\n",
    "\n",
    "    return set(genes), set(motifs)\n",
    "\n",
    "def GC_content(seq: str) -> float:\n",
    "    gc_count = sum(1 for base in seq if base in 'GCgc')\n",
    "    return gc_count / len(seq) if len(seq) > 0 else 0\n",
    "\n",
    "# pick out immediate neighborhood gene sets\n",
    "edges_8161 = G.in_edges(8161, data=\"annotations\")\n",
    "genes_8161, motifs_8161 = feature_geneandmotif_sets(8161, edges_8161)\n",
    "print(\"Neighborhood Size of f/8161:\", len(motifs_8161), \"/\", len(genes_8161), \"genes\")\n",
    "print(\"GC content of motifs associated with f/8161:\", np.mean([GC_content(motif) for motif in motifs_8161]))\n",
    "    \n",
    "edges_7030 = G.in_edges(7030, data=\"annotations\")\n",
    "genes_7030, motifs_7030 = feature_geneandmotif_sets(7030, edges_7030)\n",
    "print(\"Neighborhood Size of f/7030:\", len(motifs_7030), \"/\", len(genes_7030), \"genes\")\n",
    "print(\"GC content of motifs associated with f/7030:\", np.mean([GC_content(motif) for motif in motifs_7030]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "f9f022e7",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: ‘clusterProfiler’\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an\n",
      "R/Bioconductor package for Disease Ontology Semantic and Enrichment\n",
      "analysis. Bioinformatics. 2015, 31(4):608-609\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: ‘generics’\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:clusterProfiler’:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from ‘package:utils’:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: ‘IRanges’\n",
      "\n",
      "The following object is masked from ‘package:clusterProfiler’:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: ‘AnnotationDbi’\n",
      "\n",
      "The following object is masked from ‘package:clusterProfiler’:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "The following object is masked from ‘package:AnnotationDbi’:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from ‘package:IRanges’:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from ‘package:S4Vectors’:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from ‘package:Biobase’:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from ‘package:generics’:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/regelem_gene_sets/genes_f8161.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/regelem_gene_sets/enrich_f8161\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      "ℹ Please use `linewidth` instead.\n",
      "ℹ The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      "✖ Problematic argument:\n",
      "• by = x\n",
      "ℹ Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      "ℹ Please use tidy evaluation idioms with `aes()`.\n",
      "ℹ See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      "ℹ The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: ‘clusterProfiler’\n",
      "\n",
      "The following object is masked from ‘package:stats’:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, and Qing-Yu He. ChIPseeker: an\n",
      "R/Bioconductor package for ChIP peak annotation, comparison and\n",
      "visualization. Bioinformatics. 2015, 31(14):2382-2383\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: ‘generics’\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:clusterProfiler’:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from ‘package:utils’:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: ‘IRanges’\n",
      "\n",
      "The following object is masked from ‘package:clusterProfiler’:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: ‘AnnotationDbi’\n",
      "\n",
      "The following object is masked from ‘package:clusterProfiler’:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘dplyr’\n",
      "\n",
      "The following object is masked from ‘package:AnnotationDbi’:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from ‘package:IRanges’:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from ‘package:S4Vectors’:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from ‘package:Biobase’:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from ‘package:BiocGenerics’:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from ‘package:generics’:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/regelem_gene_sets/genes_f7030.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/regelem_gene_sets/enrich_f7030\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      "ℹ Please use `linewidth` instead.\n",
      "ℹ The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      "✖ Problematic argument:\n",
      "• by = x\n",
      "ℹ Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      "ℹ Please use tidy evaluation idioms with `aes()`.\n",
      "ℹ See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      "ℹ The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.makedirs(\"data/regelem_gene_sets\", exist_ok=True)\n",
    "\n",
    "# export gene sets\n",
    "with open(\"data/regelem_gene_sets/genes_f8161.txt\", \"w\") as f:\n",
    "    for gene in genes_8161:\n",
    "        f.write(gene + \"\\n\")\n",
    "\n",
    "with open(\"data/regelem_gene_sets/genes_f7030.txt\", \"w\") as f:\n",
    "    for gene in genes_7030:\n",
    "        f.write(gene + \"\\n\")\n",
    "\n",
    "# run GSEA in R using clusterProfiler\n",
    "os.system(\"Rscript enrich.R --gene_list data/regelem_gene_sets/genes_f8161.txt --outdir data/regelem_gene_sets/enrich_f8161\")\n",
    "os.system(\"Rscript enrich.R --gene_list data/regelem_gene_sets/genes_f7030.txt --outdir data/regelem_gene_sets/enrich_f7030\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c5d6f617",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "hsg",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
