{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "7dad1d62",
   "metadata": {},
   "source": [
    "# Analyzing Cisplatin-Binding Transcripts via GLM-SAE Knowledge Graphs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "cdfe6556",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MultiDiGraph with 7141 nodes and 4197345 edges\n"
     ]
    }
   ],
   "source": [
    "import networkx as nx\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "from networkx.readwrite import json_graph\n",
    "\n",
    "import os, json\n",
    "\n",
    "with open(\"data/cisplatin_neg45k_kg.json\") as f:\n",
    "    data = json.load(f)\n",
    "\n",
    "G: nx.MultiDiGraph = json_graph.node_link_graph(data, edges=\"edges\")\n",
    "print(G)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "7592b1e2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MultiDiGraph with 6824 nodes and 4197028 edges\n",
      "MultiDiGraph with 6823 nodes and 4156725 edges\n",
      "Average edge bundle size: 6.9515217598531\n",
      "Median edge bundle size: 1.0\n",
      "Number of edge bundles: 597959\n",
      "Max edge bundle size: 6057\n",
      "Min edge bundle size: 1\n",
      "Standard deviation of edge bundle sizes: 46.394946770635855\n",
      "---------- Filtering edges with less than 5 samples in their bundle ----------\n",
      "MultiDiGraph with 6823 nodes and 3387035 edges\n",
      "Average edge bundle size: 34.867562281243565\n",
      "Median edge bundle size: 9.0\n",
      "Number of edge bundles: 97140\n",
      "Max edge bundle size: 6057\n",
      "Min edge bundle size: 5\n",
      "Standard deviation of edge bundle sizes: 110.97611840464725\n",
      "MultiDiGraph with 5568 nodes and 3387035 edges\n"
     ]
    }
   ],
   "source": [
    "# graph is large, so lets trim down a version we can visualize\n",
    "OG = G.copy()\n",
    "\n",
    "# filter features that only respond to one token\n",
    "to_remove = [n for n in G.nodes if G.in_degree(n) == 1]\n",
    "G.remove_nodes_from(to_remove)\n",
    "print(G)\n",
    "\n",
    "# the '<cls>' token is the 'start' token and does not encode genetic info\n",
    "if G.has_node('<cls>'):\n",
    "    G.remove_node('<cls>')\n",
    "    print(G)\n",
    "\n",
    "# find average edge bundle size\n",
    "edge_sizes = {(u,v): len(G.get_edge_data(u, v)) for u, v, k in G.edges}\n",
    "avg_edge_size = sum(edge_sizes.values()) / len(edge_sizes)\n",
    "print(f\"Average edge bundle size: {avg_edge_size}\")\n",
    "print(f\"Median edge bundle size: {np.median(list(edge_sizes.values()))}\")\n",
    "print(f\"Number of edge bundles: {len(edge_sizes)}\")\n",
    "print(f\"Max edge bundle size: {max(edge_sizes.values())}\")\n",
    "print(f\"Min edge bundle size: {min(edge_sizes.values())}\")\n",
    "print(f\"Standard deviation of edge bundle sizes: {pd.Series(edge_sizes.values()).std()}\")\n",
    "\n",
    "# filter edges that have less than 5 samples in their bundle, then show new descriptive stats\n",
    "print(\"---------- Filtering edges with less than 5 samples in their bundle ----------\")\n",
    "to_remove = [key for key, size in edge_sizes.items() if size < 5]\n",
    "for ebundle in to_remove:\n",
    "    u, v = ebundle\n",
    "    keys = list(G.get_edge_data(u, v).keys())\n",
    "    for k in keys:\n",
    "        G.remove_edge(u, v, k)\n",
    "# G.remove_edges_from(to_remove)\n",
    "print(G)\n",
    "\n",
    "edge_sizes = {(u,v): len(G.get_edge_data(u, v)) for u, v, k in G.edges}\n",
    "avg_edge_size = sum(edge_sizes.values()) / len(edge_sizes)\n",
    "print(f\"Average edge bundle size: {avg_edge_size}\")\n",
    "print(f\"Median edge bundle size: {np.median(list(edge_sizes.values()))}\")\n",
    "print(f\"Number of edge bundles: {len(edge_sizes)}\")\n",
    "print(f\"Max edge bundle size: {max(edge_sizes.values())}\")\n",
    "print(f\"Min edge bundle size: {min(edge_sizes.values())}\")\n",
    "print(f\"Standard deviation of edge bundle sizes: {pd.Series(edge_sizes.values()).std()}\")\n",
    "\n",
    "# re-filter features that only respond to one token or are disconnected from the graph\n",
    "to_remove = [n for n in G.nodes if G.in_degree(n) <= 1 and type(n) == int]\n",
    "G.remove_nodes_from(to_remove)\n",
    "print(G)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7b9d1dc6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Coding graph: MultiDiGraph with 5415 nodes and 812369 edges\n",
      "Non-coding graph: MultiDiGraph with 5567 nodes and 2574666 edges\n"
     ]
    }
   ],
   "source": [
    "# use annotations to create two graphs of coding and non-coding genomic regions\n",
    "coding_regions = [e for e in G.edges(data=True, keys=True) if len(e[3][\"annotations\"]) != 0]\n",
    "non_coding_regions = [e for e in G.edges(data=True, keys=True) if len(e[3][\"annotations\"]) == 0]\n",
    "\n",
    "G_coding = nx.MultiDiGraph()\n",
    "for edge in coding_regions:\n",
    "    G_coding.add_edge(edge[0], edge[1], edge[2], **edge[3])\n",
    "G_non_coding = nx.MultiDiGraph()\n",
    "for edge in non_coding_regions:\n",
    "    G_non_coding.add_edge(edge[0], edge[1], edge[2], **edge[3])\n",
    "\n",
    "print(f\"Coding graph: {G_coding}\")\n",
    "print(f\"Non-coding graph: {G_non_coding}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "73f24293",
   "metadata": {},
   "outputs": [],
   "source": [
    "def node_color(x):\n",
    "    return \"lightblue\" if type(x) == str else \"red\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8fa86309",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- CODING SEQUENCES ---\n",
      "Top 10 nodes by PageRank:\n",
      "Node: 8161, PageRank: 0.19687593503427994\n",
      "Node: 7030, PageRank: 0.0372370455963754\n",
      "Node: 7949, PageRank: 0.014281489531611225\n",
      "Node: 1422, PageRank: 0.007482991577658241\n",
      "Node: 5984, PageRank: 0.006915740386288211\n",
      "Node: 2453, PageRank: 0.0048780476203867185\n",
      "Node: 9452, PageRank: 0.004333574960662349\n",
      "Node: 9021, PageRank: 0.003083666406978392\n",
      "Node: 835, PageRank: 0.002501853543098789\n",
      "Node: 5741, PageRank: 0.0024684622382828867\n",
      "Top 10 nodes by Out-Degree Centrality:\n",
      "Node: TTTTTT, Out-Degree Centrality: 0.25434059844846696\n",
      "Node: AAAAAA, Out-Degree Centrality: 0.2133357960842261\n",
      "Node: ATTTTT, Out-Degree Centrality: 0.18729220539342445\n",
      "Node: TTTTTA, Out-Degree Centrality: 0.17713335796084226\n",
      "Node: G, Out-Degree Centrality: 0.16715921684521612\n",
      "Node: TATTTT, Out-Degree Centrality: 0.16642039157739194\n",
      "Node: AAAAAT, Out-Degree Centrality: 0.16512744735869966\n",
      "Node: TTTTTG, Out-Degree Centrality: 0.16217214628740304\n",
      "Node: TTTTAA, Out-Degree Centrality: 0.15293683043960105\n",
      "Node: TTATTT, Out-Degree Centrality: 0.15238271148873292\n"
     ]
    }
   ],
   "source": [
    "print(\"--- CODING SEQUENCES ---\")\n",
    "\n",
    "pagerank = nx.pagerank(G_coding)\n",
    "sorted_pr = sorted(pagerank.items(), key=lambda item: item[1], reverse=True)\n",
    "print(\"Top 10 nodes by PageRank:\")\n",
    "for node, pr in sorted_pr[:10]:\n",
    "    print(f\"Node: {node}, PageRank: {pr}\")\n",
    "\n",
    "degree_centrality = nx.out_degree_centrality(G_coding)\n",
    "sorted_degree = sorted(degree_centrality.items(), key=lambda x: x[1], reverse=True)\n",
    "print(\"Top 10 nodes by Out-Degree Centrality:\")\n",
    "for node, dc in sorted_degree[:10]:\n",
    "    print(f\"Node: {node}, Out-Degree Centrality: {dc}\")\n",
    "\n",
    "#nx.draw(G_coding, pos=nx.forceatlas2_layout(G_coding), with_labels=False, node_size=10, font_size=4, node_color=[node_color(n) for n in G_coding.nodes()])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "bfff610f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--- NON-CODING SEQUENCES ---\n",
      "Top 10 nodes by PageRank:\n",
      "Node: 8161, PageRank: 0.16845245564335232\n",
      "Node: 7030, PageRank: 0.029691228481648322\n",
      "Node: 3378, PageRank: 0.012824458852998533\n",
      "Node: 4793, PageRank: 0.0095631601344086\n",
      "Node: 7949, PageRank: 0.006341539512496292\n",
      "Node: 1422, PageRank: 0.006265641335004452\n",
      "Node: 5984, PageRank: 0.00532525965976957\n",
      "Node: 1545, PageRank: 0.004422141159927955\n",
      "Node: 835, PageRank: 0.0043350835181561135\n",
      "Node: 7612, PageRank: 0.004331091308329926\n",
      "Top 10 nodes by Out-Degree Centrality:\n",
      "Node: G, Out-Degree Centrality: 1.7919511318720804\n",
      "Node: C, Out-Degree Centrality: 1.7642831476823573\n",
      "Node: A, Out-Degree Centrality: 1.6658282429033417\n",
      "Node: T, Out-Degree Centrality: 1.5449155587495509\n",
      "Node: AAAAAA, Out-Degree Centrality: 0.8841178584261589\n",
      "Node: TTTTTT, Out-Degree Centrality: 0.8514193316564859\n",
      "Node: AAAAAT, Out-Degree Centrality: 0.5979159180740209\n",
      "Node: ATTTTT, Out-Degree Centrality: 0.5725835429392742\n",
      "Node: TAAAAA, Out-Degree Centrality: 0.541501976284585\n",
      "Node: AAAATA, Out-Degree Centrality: 0.4996406755300036\n"
     ]
    }
   ],
   "source": [
    "print(\"--- NON-CODING SEQUENCES ---\")\n",
    "\n",
    "pagerank = nx.pagerank(G_non_coding)\n",
    "sorted_pr = sorted(pagerank.items(), key=lambda item: item[1], reverse=True)\n",
    "print(\"Top 10 nodes by PageRank:\")\n",
    "for node, pr in sorted_pr[:10]:\n",
    "    print(f\"Node: {node}, PageRank: {pr}\")\n",
    "\n",
    "degree_centrality = nx.out_degree_centrality(G_non_coding)\n",
    "sorted_degree = sorted(degree_centrality.items(), key=lambda x: x[1], reverse=True)\n",
    "print(\"Top 10 nodes by Out-Degree Centrality:\")\n",
    "for node, dc in sorted_degree[:10]:\n",
    "    print(f\"Node: {node}, Out-Degree Centrality: {dc}\")\n",
    "\n",
    "\n",
    "#nx.draw(G_non_coding, pos=nx.forceatlas2_layout(G_non_coding), with_labels=False, node_size=10, font_size=4, node_color=[node_color(n) for n in G_non_coding.nodes()])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "09a5585c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nodes ranked by PageRank:\n",
      "8161: 0.10874592750416887\n",
      "7030: 0.021950368456290115\n",
      "3378: 0.007113011140264806\n",
      "4793: 0.005248423700916205\n",
      "7949: 0.004543801336913997\n",
      "5984: 0.004379874340400829\n",
      "1422: 0.00405684001443536\n",
      "2453: 0.0032858460917511473\n",
      "7612: 0.002891826775626524\n",
      "835: 0.0028105569144162477\n"
     ]
    }
   ],
   "source": [
    "# COMPARE TO ORIGINAL GRAPH\n",
    "# use page rank to measure centrality\n",
    "pagerank = nx.pagerank(OG)\n",
    "sorted_pagerank = sorted(pagerank.items(), key=lambda x: x[1], reverse=True)\n",
    "\n",
    "print(\"Nodes ranked by PageRank:\")\n",
    "for node, score in sorted_pagerank[:10]:  # top 10 nodes\n",
    "    print(f\"{node}: {score}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "d63c47a9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Nodes ranked by Degree Centrality:\n",
      "<cls>: 5.644677871148459\n",
      "G: 1.6274509803921569\n",
      "C: 1.5823529411764705\n",
      "A: 1.4928571428571427\n",
      "T: 1.3866946778711484\n",
      "TTTTTT: 0.8684873949579831\n",
      "AAAAAA: 0.8638655462184873\n",
      "ATTTTT: 0.6273109243697479\n",
      "AAAAAT: 0.6194677871148458\n",
      "TAAAAA: 0.5546218487394957\n"
     ]
    }
   ],
   "source": [
    "# COMPARE TO ORIGINAL GRAPH\n",
    "# compare with degree centrality for important tokens\n",
    "degree_centrality = nx.out_degree_centrality(OG)\n",
    "sorted_degree = sorted(degree_centrality.items(), key=lambda x: x[1], reverse=True)\n",
    "\n",
    "print(\"Nodes ranked by Degree Centrality:\")\n",
    "for node, score in sorted_degree[:10]:  # top 10 nodes\n",
    "    print(f\"{node}: {score}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "e2044dad",
   "metadata": {},
   "outputs": [],
   "source": [
    "# export to cytoscape for visualization\n",
    "# Didn't work for the large graph, so just keeping for documentation\n",
    "for node in OG.nodes():\n",
    "    OG.nodes[node]['id'] = node\n",
    "\n",
    "cyto_data = nx.readwrite.json_graph.cytoscape_data(OG)\n",
    "with open(\"data/cisplatin_neg45k_kg_cyto.json\", \"w\") as f:\n",
    "    json.dump(cyto_data, f)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c7ae8834",
   "metadata": {},
   "source": [
    "# Using Knowledge Graph Patterns to Create Gene Sets\n",
    "\n",
    "Note the strong centrality in multiple segments of the graph for features 3378, 407, and 4793, as well as typical base pairing hairpin motifs with high GC content in the outbound central nodes. \n",
    "\n",
    "By gathering data from the edges associated with each major feature, we can create gene sets based on GLM feature output. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "45a04b4b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Note that our non-coding graph lacks annotations by definition, so we will only create gene sets for coding regions\n",
    "# central feat 0\n",
    "best_feature = sorted_pagerank[0][0]\n",
    "best_feature2 = sorted_pagerank[1][0]\n",
    "best_feature3 = sorted_pagerank[2][0]\n",
    "\n",
    "edges = G_coding.in_edges(best_feature, data=\"annotations\")\n",
    "\n",
    "genes = []\n",
    "motifs = []\n",
    "for u, v, annotations in edges:\n",
    "    for key, val in annotations.items():\n",
    "        genes.append(val[\"gene_id\"])\n",
    "        motifs.append(u)\n",
    "\n",
    "os.makedirs(\"data/neg_gene_sets\", exist_ok=True)\n",
    "\n",
    "with open(f\"data/neg_gene_sets/feature_{best_feature}_genes.txt\", \"w\") as f:\n",
    "    f.write(\"\\n\".join(set(genes)))\n",
    "\n",
    "with open(f\"data/neg_gene_sets/feature_{best_feature}_motifs.txt\", \"w\") as f:\n",
    "    f.write(\"\\n\".join(set(motifs)))\n",
    "\n",
    "# 407\n",
    "edges = G_coding.in_edges(best_feature2, data=\"annotations\")\n",
    "\n",
    "genes = []\n",
    "motifs = []\n",
    "for u, v, annotations in edges:\n",
    "    for key, val in annotations.items():\n",
    "        genes.append(val[\"gene_id\"])\n",
    "        motifs.append(u)\n",
    "\n",
    "with open(f\"data/neg_gene_sets/feature_{best_feature2}_genes.txt\", \"w\") as f:\n",
    "    f.write(\"\\n\".join(set(genes)))\n",
    "\n",
    "with open(f\"data/neg_gene_sets/feature_{best_feature2}_motifs.txt\", \"w\") as f:\n",
    "    f.write(\"\\n\".join(set(motifs)))\n",
    "\n",
    "\n",
    "# 4793\n",
    "edges = G_coding.in_edges(best_feature3, data=\"annotations\")\n",
    "\n",
    "genes = []\n",
    "motifs = []\n",
    "for u, v, annotations in edges:\n",
    "    for key, val in annotations.items():\n",
    "        genes.append(val[\"gene_id\"])\n",
    "        motifs.append(u)\n",
    "\n",
    "with open(f\"data/neg_gene_sets/feature_{best_feature3}_genes.txt\", \"w\") as f:\n",
    "    f.write(\"\\n\".join(set(genes)))\n",
    "\n",
    "with open(f\"data/neg_gene_sets/feature_{best_feature3}_motifs.txt\", \"w\") as f:\n",
    "    f.write(\"\\n\".join(set(motifs)))\n",
    "\n",
    "\n",
    "# overall gene sets for comparison\n",
    "edges = G_coding.edges(data=\"annotations\")\n",
    "genes = []\n",
    "motifs = []\n",
    "for u, v, annotations in edges:\n",
    "    for key, val in annotations.items():\n",
    "        genes.append(val[\"gene_id\"])\n",
    "        motifs.append(u)\n",
    "with open(\"data/neg_gene_sets/all_coding_genes.txt\", \"w\") as f:\n",
    "    f.write(\"\\n\".join(set(genes)))\n",
    "with open(\"data/neg_gene_sets/all_coding_motifs.txt\", \"w\") as f:\n",
    "    f.write(\"\\n\".join(set(motifs)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "ce231f6f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# free up memory\n",
    "del G\n",
    "del G_coding\n",
    "del G_non_coding\n",
    "del cyto_data\n",
    "del edges\n",
    "del genes\n",
    "del motifs\n",
    "del sorted_pagerank\n",
    "del sorted_pr"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e904fc4d",
   "metadata": {},
   "source": [
    "# Community Detection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "97491b57",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 43 communities\n"
     ]
    }
   ],
   "source": [
    "# Louvain\n",
    "communities = nx.community.louvain_communities(OG.to_undirected(), resolution=1.5)\n",
    "print(f\"Found {len(communities)} communities\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "46879873",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Community 0:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 5746, In-Degree Centrality: 35124.0\n",
      "  Node: <cls>, In-Degree Centrality: 0.0\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: <cls>, Out-Degree Centrality: 35124.0\n",
      "  Node: 5746, Out-Degree Centrality: 0.0\n",
      "\n",
      "Community 1:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 7030, In-Degree Centrality: 323.2313432835821\n",
      "  Node: 2471, In-Degree Centrality: 6.384328358208955\n",
      "  Node: 720, In-Degree Centrality: 5.527985074626866\n",
      "  Node: 1003, In-Degree Centrality: 4.227611940298507\n",
      "  Node: 954, In-Degree Centrality: 4.210820895522388\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: AGAAAA, Out-Degree Centrality: 5.975746268656716\n",
      "  Node: TTTTTA, Out-Degree Centrality: 5.656716417910448\n",
      "  Node: AAGAAA, Out-Degree Centrality: 5.5783582089552235\n",
      "  Node: TTTTAA, Out-Degree Centrality: 5.317164179104478\n",
      "  Node: ATTTTT, Out-Degree Centrality: 5.255597014925373\n",
      "\n",
      "Community 2:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 1845, In-Degree Centrality: 9.22936507936508\n",
      "  Node: 7949, In-Degree Centrality: 8.313492063492063\n",
      "  Node: 1422, In-Degree Centrality: 6.447619047619048\n",
      "  Node: 407, In-Degree Centrality: 6.292063492063492\n",
      "  Node: 7612, In-Degree Centrality: 5.68968253968254\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: G, Out-Degree Centrality: 6.395238095238096\n",
      "  Node: C, Out-Degree Centrality: 6.241269841269841\n",
      "  Node: A, Out-Degree Centrality: 5.957936507936508\n",
      "  Node: T, Out-Degree Centrality: 5.7063492063492065\n",
      "  Node: CAGCAG, Out-Degree Centrality: 0.3047619047619048\n",
      "\n",
      "Community 3:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 1389, In-Degree Centrality: 46.42028985507246\n",
      "  Node: 2216, In-Degree Centrality: 28.73913043478261\n",
      "  Node: 9040, In-Degree Centrality: 11.840579710144928\n",
      "  Node: 9681, In-Degree Centrality: 11.333333333333334\n",
      "  Node: 9748, In-Degree Centrality: 8.768115942028986\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: GGGATT, Out-Degree Centrality: 16.797101449275363\n",
      "  Node: ACAGGC, Out-Degree Centrality: 15.434782608695652\n",
      "  Node: ACAGAG, Out-Degree Centrality: 11.391304347826088\n",
      "  Node: GGAGTT, Out-Degree Centrality: 10.898550724637682\n",
      "  Node: TTTCAC, Out-Degree Centrality: 9.434782608695652\n",
      "\n",
      "Community 4:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 4814, In-Degree Centrality: 174.72857142857143\n",
      "  Node: 9339, In-Degree Centrality: 68.62857142857143\n",
      "  Node: 1781, In-Degree Centrality: 66.22857142857143\n",
      "  Node: 1096, In-Degree Centrality: 12.385714285714286\n",
      "  Node: 2726, In-Degree Centrality: 9.842857142857143\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: AAAAAA, Out-Degree Centrality: 84.31428571428572\n",
      "  Node: AAAAAT, Out-Degree Centrality: 52.628571428571426\n",
      "  Node: AAAATA, Out-Degree Centrality: 39.528571428571425\n",
      "  Node: TAAAAA, Out-Degree Centrality: 32.31428571428572\n",
      "  Node: AAAATT, Out-Degree Centrality: 27.0\n",
      "\n",
      "Community 5:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 4793, In-Degree Centrality: 24.879166666666666\n",
      "  Node: 5458, In-Degree Centrality: 13.237499999999999\n",
      "  Node: 1895, In-Degree Centrality: 3.8791666666666664\n",
      "  Node: 7940, In-Degree Centrality: 2.7041666666666666\n",
      "  Node: 6480, In-Degree Centrality: 2.158333333333333\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: TGTGTG, Out-Degree Centrality: 5.683333333333334\n",
      "  Node: GTGTGT, Out-Degree Centrality: 3.995833333333333\n",
      "  Node: TATATA, Out-Degree Centrality: 2.6083333333333334\n",
      "  Node: ATATAT, Out-Degree Centrality: 2.4791666666666665\n",
      "  Node: CACACA, Out-Degree Centrality: 1.625\n",
      "\n",
      "Community 6:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 4831, In-Degree Centrality: 492.52941176470586\n",
      "  Node: 559, In-Degree Centrality: 17.294117647058822\n",
      "  Node: 2656, In-Degree Centrality: 14.76470588235294\n",
      "  Node: 1516, In-Degree Centrality: 10.51470588235294\n",
      "  Node: 1613, In-Degree Centrality: 8.882352941176471\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: TTTTTT, Out-Degree Centrality: 89.16176470588235\n",
      "  Node: TTTCTT, Out-Degree Centrality: 43.955882352941174\n",
      "  Node: CTTTTT, Out-Degree Centrality: 42.838235294117645\n",
      "  Node: TATTTT, Out-Degree Centrality: 41.19117647058823\n",
      "  Node: TTTTCT, Out-Degree Centrality: 38.294117647058826\n",
      "\n",
      "Community 7:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 1636, In-Degree Centrality: 20.787096774193547\n",
      "  Node: 6347, In-Degree Centrality: 16.44516129032258\n",
      "  Node: 74, In-Degree Centrality: 12.503225806451614\n",
      "  Node: 7836, In-Degree Centrality: 9.761290322580646\n",
      "  Node: 2001, In-Degree Centrality: 8.825806451612904\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: TGGGAT, Out-Degree Centrality: 7.735483870967742\n",
      "  Node: TGGAGT, Out-Degree Centrality: 6.780645161290322\n",
      "  Node: TTGAGA, Out-Degree Centrality: 6.245161290322581\n",
      "  Node: TGAGAC, Out-Degree Centrality: 6.2\n",
      "  Node: CTCACT, Out-Degree Centrality: 5.664516129032258\n",
      "\n",
      "Community 8:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 6445, In-Degree Centrality: 9.264797507788161\n",
      "  Node: 4554, In-Degree Centrality: 6.383177570093458\n",
      "  Node: 9656, In-Degree Centrality: 5.5607476635514015\n",
      "  Node: 2416, In-Degree Centrality: 4.339563862928348\n",
      "  Node: 7979, In-Degree Centrality: 4.242990654205608\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: GGGAGG, Out-Degree Centrality: 5.591900311526479\n",
      "  Node: GAGGCA, Out-Degree Centrality: 3.5856697819314642\n",
      "  Node: GGCAGG, Out-Degree Centrality: 3.398753894080997\n",
      "  Node: GCAGGA, Out-Degree Centrality: 2.588785046728972\n",
      "  Node: AGCACT, Out-Degree Centrality: 2.5171339563862927\n",
      "\n",
      "Community 9:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 8453, In-Degree Centrality: 23.98507462686567\n",
      "  Node: 5840, In-Degree Centrality: 18.776119402985074\n",
      "  Node: 3381, In-Degree Centrality: 17.37313432835821\n",
      "  Node: 3566, In-Degree Centrality: 12.432835820895523\n",
      "  Node: 1037, In-Degree Centrality: 12.134328358208956\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: CTGGGA, Out-Degree Centrality: 23.119402985074625\n",
      "  Node: CTGAGG, Out-Degree Centrality: 17.880597014925375\n",
      "  Node: GGTTGC, Out-Degree Centrality: 7.985074626865671\n",
      "  Node: CTGAGA, Out-Degree Centrality: 6.17910447761194\n",
      "  Node: CTGAGT, Out-Degree Centrality: 5.343283582089552\n",
      "\n",
      "Community 10:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 8358, In-Degree Centrality: 17.886666666666667\n",
      "  Node: 5781, In-Degree Centrality: 12.093333333333334\n",
      "  Node: 8720, In-Degree Centrality: 10.24\n",
      "  Node: 1108, In-Degree Centrality: 10.006666666666668\n",
      "  Node: 4524, In-Degree Centrality: 7.48\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: AGGCTG, Out-Degree Centrality: 14.746666666666668\n",
      "  Node: CCCAGG, Out-Degree Centrality: 9.573333333333334\n",
      "  Node: AGGCAG, Out-Degree Centrality: 8.126666666666667\n",
      "  Node: CCTGGG, Out-Degree Centrality: 7.773333333333333\n",
      "  Node: CAGCTA, Out-Degree Centrality: 7.16\n",
      "\n",
      "Community 11:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 8853, In-Degree Centrality: 31.32051282051282\n",
      "  Node: 8847, In-Degree Centrality: 29.51282051282051\n",
      "  Node: 6885, In-Degree Centrality: 12.935897435897436\n",
      "  Node: 8337, In-Degree Centrality: 12.923076923076923\n",
      "  Node: 6506, In-Degree Centrality: 8.192307692307692\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: AATTAG, Out-Degree Centrality: 10.923076923076923\n",
      "  Node: CTGGGC, Out-Degree Centrality: 9.846153846153847\n",
      "  Node: GTGAAA, Out-Degree Centrality: 7.9743589743589745\n",
      "  Node: AACATG, Out-Degree Centrality: 7.794871794871795\n",
      "  Node: CTGGGT, Out-Degree Centrality: 7.256410256410256\n",
      "\n",
      "Community 12:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 8356, In-Degree Centrality: 22.977777777777778\n",
      "  Node: 6485, In-Degree Centrality: 16.211111111111112\n",
      "  Node: 1271, In-Degree Centrality: 15.877777777777778\n",
      "  Node: 1497, In-Degree Centrality: 11.933333333333334\n",
      "  Node: 3255, In-Degree Centrality: 9.866666666666667\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: CACTGC, Out-Degree Centrality: 16.755555555555556\n",
      "  Node: GTGAGC, Out-Degree Centrality: 14.488888888888889\n",
      "  Node: GAGCCA, Out-Degree Centrality: 10.977777777777778\n",
      "  Node: ACTCCA, Out-Degree Centrality: 8.88888888888889\n",
      "  Node: CACCAT, Out-Degree Centrality: 8.855555555555556\n",
      "\n",
      "Community 13:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 1261, In-Degree Centrality: 36.932432432432435\n",
      "  Node: 4171, In-Degree Centrality: 26.54054054054054\n",
      "  Node: 7674, In-Degree Centrality: 20.256756756756758\n",
      "  Node: 860, In-Degree Centrality: 17.756756756756758\n",
      "  Node: 1666, In-Degree Centrality: 11.081081081081082\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: CAGCCT, Out-Degree Centrality: 28.783783783783786\n",
      "  Node: CCAGCC, Out-Degree Centrality: 16.83783783783784\n",
      "  Node: GGATTA, Out-Degree Centrality: 16.256756756756758\n",
      "  Node: GCACTC, Out-Degree Centrality: 10.986486486486488\n",
      "  Node: TGCACT, Out-Degree Centrality: 10.95945945945946\n",
      "\n",
      "Community 14:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 6912, In-Degree Centrality: 31.850467289719624\n",
      "  Node: 7911, In-Degree Centrality: 11.130841121495326\n",
      "  Node: 4588, In-Degree Centrality: 11.093457943925232\n",
      "  Node: 2496, In-Degree Centrality: 10.990654205607475\n",
      "  Node: 6795, In-Degree Centrality: 9.672897196261681\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: GCCTGT, Out-Degree Centrality: 10.299065420560748\n",
      "  Node: AATCCC, Out-Degree Centrality: 10.093457943925232\n",
      "  Node: TTGGGA, Out-Degree Centrality: 9.785046728971961\n",
      "  Node: GTGGCT, Out-Degree Centrality: 8.757009345794392\n",
      "  Node: GTGGTG, Out-Degree Centrality: 8.42056074766355\n",
      "\n",
      "Community 15:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 8161, In-Degree Centrality: 484.08926875593545\n",
      "  Node: 5961, In-Degree Centrality: 1.0911680911680912\n",
      "  Node: 1838, In-Degree Centrality: 0.5887939221272555\n",
      "  Node: 2971, In-Degree Centrality: 0.5242165242165242\n",
      "  Node: 1658, In-Degree Centrality: 0.21462488129154797\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: ACATTT, Out-Degree Centrality: 1.3475783475783476\n",
      "  Node: TTCCTT, Out-Degree Centrality: 1.1671415004748338\n",
      "  Node: TTCTCT, Out-Degree Centrality: 1.1528964862298197\n",
      "  Node: TTCATT, Out-Degree Centrality: 1.1329534662867997\n",
      "  Node: TTAATT, Out-Degree Centrality: 1.123456790123457\n",
      "\n",
      "Community 16:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 700, In-Degree Centrality: 22.678899082568808\n",
      "  Node: 3456, In-Degree Centrality: 18.57798165137615\n",
      "  Node: 1545, In-Degree Centrality: 14.357798165137616\n",
      "  Node: 6122, In-Degree Centrality: 7.660550458715597\n",
      "  Node: 7597, In-Degree Centrality: 7.220183486238533\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: GCCTCA, Out-Degree Centrality: 7.7431192660550465\n",
      "  Node: CCTGAC, Out-Degree Centrality: 5.541284403669725\n",
      "  Node: ACCTCA, Out-Degree Centrality: 5.467889908256881\n",
      "  Node: TGATCT, Out-Degree Centrality: 4.825688073394495\n",
      "  Node: CTCAAG, Out-Degree Centrality: 4.110091743119266\n",
      "\n",
      "Community 17:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 1052, In-Degree Centrality: 24.807339449541285\n",
      "  Node: 1418, In-Degree Centrality: 21.394495412844037\n",
      "  Node: 4742, In-Degree Centrality: 21.28440366972477\n",
      "  Node: 6719, In-Degree Centrality: 20.63302752293578\n",
      "  Node: 8024, In-Degree Centrality: 10.541284403669726\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: GCTGGG, Out-Degree Centrality: 14.577981651376147\n",
      "  Node: ATTACA, Out-Degree Centrality: 11.568807339449542\n",
      "  Node: GATTAC, Out-Degree Centrality: 10.77981651376147\n",
      "  Node: TCCCAA, Out-Degree Centrality: 10.119266055045872\n",
      "  Node: AGCTGG, Out-Degree Centrality: 9.844036697247708\n",
      "\n",
      "Community 18:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 329, In-Degree Centrality: 29.050847457627118\n",
      "  Node: 6992, In-Degree Centrality: 17.016949152542374\n",
      "  Node: 4281, In-Degree Centrality: 14.5\n",
      "  Node: 2494, In-Degree Centrality: 12.567796610169491\n",
      "  Node: 8857, In-Degree Centrality: 8.305084745762711\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: TGCCTC, Out-Degree Centrality: 9.203389830508474\n",
      "  Node: GTCTCA, Out-Degree Centrality: 7.11864406779661\n",
      "  Node: TGGCCA, Out-Degree Centrality: 6.728813559322034\n",
      "  Node: TCTCTA, Out-Degree Centrality: 6.364406779661017\n",
      "  Node: TGTCTC, Out-Degree Centrality: 6.1440677966101696\n",
      "\n",
      "Community 19:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 7622, In-Degree Centrality: 36.825\n",
      "  Node: 10226, In-Degree Centrality: 12.8125\n",
      "  Node: 8939, In-Degree Centrality: 12.4375\n",
      "  Node: 7126, In-Degree Centrality: 11.775\n",
      "  Node: 8258, In-Degree Centrality: 11.700000000000001\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: GCAGTG, Out-Degree Centrality: 16.275000000000002\n",
      "  Node: GCCACC, Out-Degree Centrality: 13.0\n",
      "  Node: GATCAC, Out-Degree Centrality: 10.137500000000001\n",
      "  Node: GCCACT, Out-Degree Centrality: 9.950000000000001\n",
      "  Node: ACCAGC, Out-Degree Centrality: 8.8625\n",
      "\n",
      "Community 20:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 7990, In-Degree Centrality: 15.67487684729064\n",
      "  Node: 5234, In-Degree Centrality: 13.94088669950739\n",
      "  Node: 9010, In-Degree Centrality: 8.108374384236454\n",
      "  Node: 2957, In-Degree Centrality: 7.039408866995074\n",
      "  Node: 1179, In-Degree Centrality: 4.54679802955665\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: CCCAGC, Out-Degree Centrality: 9.128078817733991\n",
      "  Node: AGCCAC, Out-Degree Centrality: 4.990147783251231\n",
      "  Node: GGCATG, Out-Degree Centrality: 4.620689655172414\n",
      "  Node: TCCAGC, Out-Degree Centrality: 4.41871921182266\n",
      "  Node: CCTGGC, Out-Degree Centrality: 4.41871921182266\n",
      "\n",
      "Community 21:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 7664, In-Degree Centrality: 13.070707070707071\n",
      "  Node: 527, In-Degree Centrality: 11.989898989898991\n",
      "  Node: 7565, In-Degree Centrality: 11.626262626262628\n",
      "  Node: 6069, In-Degree Centrality: 9.545454545454547\n",
      "  Node: 6006, In-Degree Centrality: 7.3232323232323235\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: CACCTG, Out-Degree Centrality: 8.91919191919192\n",
      "  Node: TAATCC, Out-Degree Centrality: 7.93939393939394\n",
      "  Node: TGCCTG, Out-Degree Centrality: 7.838383838383839\n",
      "  Node: GCTAAT, Out-Degree Centrality: 7.737373737373738\n",
      "  Node: TGCCCA, Out-Degree Centrality: 7.242424242424243\n",
      "\n",
      "Community 22:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 10234, In-Degree Centrality: 25.7741935483871\n",
      "  Node: 4034, In-Degree Centrality: 12.591397849462366\n",
      "  Node: 6696, In-Degree Centrality: 10.78494623655914\n",
      "  Node: 309, In-Degree Centrality: 9.78494623655914\n",
      "  Node: 8088, In-Degree Centrality: 9.731182795698926\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: GGTCTC, Out-Degree Centrality: 8.10752688172043\n",
      "  Node: GACCTC, Out-Degree Centrality: 6.67741935483871\n",
      "  Node: GATCTC, Out-Degree Centrality: 6.107526881720431\n",
      "  Node: GATTCT, Out-Degree Centrality: 5.849462365591398\n",
      "  Node: AACCCC, Out-Degree Centrality: 5.774193548387097\n",
      "\n",
      "Community 23:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 7714, In-Degree Centrality: 28.941176470588236\n",
      "  Node: 9100, In-Degree Centrality: 26.223529411764705\n",
      "  Node: 8958, In-Degree Centrality: 15.0\n",
      "  Node: 1168, In-Degree Centrality: 12.847058823529412\n",
      "  Node: 4725, In-Degree Centrality: 9.670588235294117\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: CCAGGC, Out-Degree Centrality: 16.32941176470588\n",
      "  Node: CAGGCT, Out-Degree Centrality: 15.847058823529412\n",
      "  Node: TGGTCT, Out-Degree Centrality: 10.376470588235295\n",
      "  Node: TCACTG, Out-Degree Centrality: 9.51764705882353\n",
      "  Node: TGTTGC, Out-Degree Centrality: 6.235294117647059\n",
      "\n",
      "Community 24:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 526, In-Degree Centrality: 33.44\n",
      "  Node: 8660, In-Degree Centrality: 31.94\n",
      "  Node: 8300, In-Degree Centrality: 18.82\n",
      "  Node: 1582, In-Degree Centrality: 17.080000000000002\n",
      "  Node: 7069, In-Degree Centrality: 15.56\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: GCCTCC, Out-Degree Centrality: 35.64\n",
      "  Node: ACTGCA, Out-Degree Centrality: 24.04\n",
      "  Node: CCTCAG, Out-Degree Centrality: 21.02\n",
      "  Node: GTGATC, Out-Degree Centrality: 16.98\n",
      "  Node: ACCTCC, Out-Degree Centrality: 11.5\n",
      "\n",
      "Community 25:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 5871, In-Degree Centrality: 37.175000000000004\n",
      "  Node: 6784, In-Degree Centrality: 9.975000000000001\n",
      "  Node: 2857, In-Degree Centrality: 7.18125\n",
      "  Node: 1876, In-Degree Centrality: 5.331250000000001\n",
      "  Node: 4204, In-Degree Centrality: 5.2562500000000005\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: TACAGG, Out-Degree Centrality: 8.918750000000001\n",
      "  Node: GGTGGC, Out-Degree Centrality: 6.65\n",
      "  Node: CAGTGG, Out-Degree Centrality: 5.96875\n",
      "  Node: AGTAGC, Out-Degree Centrality: 5.3375\n",
      "  Node: AGTGGC, Out-Degree Centrality: 4.99375\n",
      "\n",
      "Community 26:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 9291, In-Degree Centrality: 68.04761904761904\n",
      "  Node: 6753, In-Degree Centrality: 40.80952380952381\n",
      "  Node: 1114, In-Degree Centrality: 16.333333333333332\n",
      "  Node: 9021, In-Degree Centrality: 13.523809523809524\n",
      "  Node: 3993, In-Degree Centrality: 10.142857142857142\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: TTACAG, Out-Degree Centrality: 56.238095238095234\n",
      "  Node: CTGTAA, Out-Degree Centrality: 54.42857142857142\n",
      "  Node: CTACAG, Out-Degree Centrality: 13.476190476190476\n",
      "  Node: TTATAG, Out-Degree Centrality: 9.095238095238095\n",
      "  Node: CTATAA, Out-Degree Centrality: 6.619047619047619\n",
      "\n",
      "Community 27:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 8911, In-Degree Centrality: 52.80645161290323\n",
      "  Node: 3071, In-Degree Centrality: 24.7741935483871\n",
      "  Node: 4914, In-Degree Centrality: 16.827956989247312\n",
      "  Node: 3407, In-Degree Centrality: 15.688172043010754\n",
      "  Node: 5312, In-Degree Centrality: 15.06451612903226\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: CAGGAG, Out-Degree Centrality: 16.505376344086024\n",
      "  Node: CCTGTA, Out-Degree Centrality: 15.473118279569894\n",
      "  Node: TGGGAG, Out-Degree Centrality: 14.860215053763442\n",
      "  Node: ATCCCA, Out-Degree Centrality: 12.655913978494624\n",
      "  Node: GCTGAG, Out-Degree Centrality: 12.344086021505378\n",
      "\n",
      "Community 28:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 4955, In-Degree Centrality: 76.69565217391305\n",
      "  Node: 2069, In-Degree Centrality: 39.08695652173913\n",
      "  Node: 1150, In-Degree Centrality: 38.34782608695652\n",
      "  Node: 6801, In-Degree Centrality: 1.391304347826087\n",
      "  Node: 2542, In-Degree Centrality: 0.3913043478260869\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: AGTGCT, Out-Degree Centrality: 35.26086956521739\n",
      "  Node: CCCAAA, Out-Degree Centrality: 34.608695652173914\n",
      "  Node: GTGCTG, Out-Degree Centrality: 32.95652173913044\n",
      "  Node: ATGTTG, Out-Degree Centrality: 29.608695652173914\n",
      "  Node: GTGTTG, Out-Degree Centrality: 6.869565217391304\n",
      "\n",
      "Community 29:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 7222, In-Degree Centrality: 71.5\n",
      "  Node: 5994, In-Degree Centrality: 70.25\n",
      "  Node: 4669, In-Degree Centrality: 7.416666666666666\n",
      "  Node: 7793, In-Degree Centrality: 3.333333333333333\n",
      "  Node: 7819, In-Degree Centrality: 0.3333333333333333\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: GAGTGC, Out-Degree Centrality: 69.08333333333333\n",
      "  Node: AAGTGC, Out-Degree Centrality: 66.25\n",
      "  Node: GAGTAC, Out-Degree Centrality: 6.916666666666666\n",
      "  Node: GAATGC, Out-Degree Centrality: 4.083333333333333\n",
      "  Node: GAGTGT, Out-Degree Centrality: 3.9166666666666665\n",
      "\n",
      "Community 30:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 5002, In-Degree Centrality: 49.25\n",
      "  Node: 653, In-Degree Centrality: 14.392857142857142\n",
      "  Node: 2453, In-Degree Centrality: 11.392857142857142\n",
      "  Node: 4570, In-Degree Centrality: 2.25\n",
      "  Node: 5769, In-Degree Centrality: 0.3571428571428571\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: GGTTCA, Out-Degree Centrality: 23.57142857142857\n",
      "  Node: CGATTC, Out-Degree Centrality: 8.75\n",
      "  Node: CAATTC, Out-Degree Centrality: 8.071428571428571\n",
      "  Node: TGATTC, Out-Degree Centrality: 6.571428571428571\n",
      "  Node: GCTCAA, Out-Degree Centrality: 6.035714285714286\n",
      "\n",
      "Community 31:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 6254, In-Degree Centrality: 44.815789473684205\n",
      "  Node: 4461, In-Degree Centrality: 15.723684210526315\n",
      "  Node: 5315, In-Degree Centrality: 13.421052631578947\n",
      "  Node: 8886, In-Degree Centrality: 11.31578947368421\n",
      "  Node: 5437, In-Degree Centrality: 7.328947368421052\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: CCTGCC, Out-Degree Centrality: 11.105263157894736\n",
      "  Node: CTGCCT, Out-Degree Centrality: 8.894736842105262\n",
      "  Node: ATTCTC, Out-Degree Centrality: 8.43421052631579\n",
      "  Node: GCAACC, Out-Degree Centrality: 7.644736842105263\n",
      "  Node: TCAAGC, Out-Degree Centrality: 7.421052631578947\n",
      "\n",
      "Community 32:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 7320, In-Degree Centrality: 29.655913978494628\n",
      "  Node: 5022, In-Degree Centrality: 29.48387096774194\n",
      "  Node: 10066, In-Degree Centrality: 13.376344086021506\n",
      "  Node: 4660, In-Degree Centrality: 11.365591397849464\n",
      "  Node: 7618, In-Degree Centrality: 11.053763440860216\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: CCAGCT, Out-Degree Centrality: 11.580645161290324\n",
      "  Node: GTAATC, Out-Degree Centrality: 11.17204301075269\n",
      "  Node: GAGGCT, Out-Degree Centrality: 10.645161290322582\n",
      "  Node: CTAATT, Out-Degree Centrality: 8.870967741935484\n",
      "  Node: TGGCTC, Out-Degree Centrality: 7.580645161290323\n",
      "\n",
      "Community 33:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 8097, In-Degree Centrality: 56.629629629629626\n",
      "  Node: 1579, In-Degree Centrality: 32.407407407407405\n",
      "  Node: 2044, In-Degree Centrality: 31.11111111111111\n",
      "  Node: 9138, In-Degree Centrality: 14.0\n",
      "  Node: 3995, In-Degree Centrality: 0.4444444444444444\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: CACTTT, Out-Degree Centrality: 29.962962962962962\n",
      "  Node: CACTCC, Out-Degree Centrality: 28.296296296296294\n",
      "  Node: GAGTCT, Out-Degree Centrality: 16.962962962962962\n",
      "  Node: CTCTCT, Out-Degree Centrality: 14.814814814814813\n",
      "  Node: CACTCT, Out-Degree Centrality: 14.592592592592592\n",
      "\n",
      "Community 34:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 3384, In-Degree Centrality: 35.111111111111114\n",
      "  Node: 9263, In-Degree Centrality: 17.3\n",
      "  Node: 8940, In-Degree Centrality: 14.700000000000001\n",
      "  Node: 1490, In-Degree Centrality: 12.6\n",
      "  Node: 2591, In-Degree Centrality: 12.033333333333333\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: GGCTCA, Out-Degree Centrality: 14.711111111111112\n",
      "  Node: TGAGCC, Out-Degree Centrality: 14.511111111111111\n",
      "  Node: TGAACC, Out-Degree Centrality: 7.677777777777778\n",
      "  Node: GTGCAG, Out-Degree Centrality: 7.177777777777778\n",
      "  Node: CTGCAA, Out-Degree Centrality: 6.955555555555556\n",
      "\n",
      "Community 35:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 4494, In-Degree Centrality: 18.156626506024097\n",
      "  Node: 4195, In-Degree Centrality: 14.180722891566266\n",
      "  Node: 288, In-Degree Centrality: 8.765060240963855\n",
      "  Node: 7571, In-Degree Centrality: 7.572289156626506\n",
      "  Node: 2346, In-Degree Centrality: 7.46987951807229\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: AGCCTG, Out-Degree Centrality: 8.078313253012048\n",
      "  Node: CCACTG, Out-Degree Centrality: 5.813253012048193\n",
      "  Node: AGTGAG, Out-Degree Centrality: 5.644578313253012\n",
      "  Node: CCACCA, Out-Degree Centrality: 5.5120481927710845\n",
      "  Node: AGACCA, Out-Degree Centrality: 4.843373493975903\n",
      "\n",
      "Community 36:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 8204, In-Degree Centrality: 79.2\n",
      "  Node: 4023, In-Degree Centrality: 25.88\n",
      "  Node: 1441, In-Degree Centrality: 23.44\n",
      "  Node: 6391, In-Degree Centrality: 16.240000000000002\n",
      "  Node: 2333, In-Degree Centrality: 7.92\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: TTGTAT, Out-Degree Centrality: 30.28\n",
      "  Node: TTTGTA, Out-Degree Centrality: 27.0\n",
      "  Node: TGTATT, Out-Degree Centrality: 26.64\n",
      "  Node: GTATTT, Out-Degree Centrality: 25.560000000000002\n",
      "  Node: TTAGTA, Out-Degree Centrality: 25.52\n",
      "\n",
      "Community 37:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 3568, In-Degree Centrality: 30.4\n",
      "  Node: 8992, In-Degree Centrality: 27.9\n",
      "  Node: 5984, In-Degree Centrality: 19.53333333333333\n",
      "  Node: 8102, In-Degree Centrality: 6.833333333333333\n",
      "  Node: 472, In-Degree Centrality: 0.4666666666666667\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: AGTGCA, Out-Degree Centrality: 27.8\n",
      "  Node: AGTTCA, Out-Degree Centrality: 11.333333333333334\n",
      "  Node: AGTTTG, Out-Degree Centrality: 10.0\n",
      "  Node: AGTTCG, Out-Degree Centrality: 9.066666666666666\n",
      "  Node: AATTCA, Out-Degree Centrality: 5.6\n",
      "\n",
      "Community 38:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 4014, In-Degree Centrality: 20.915662650602407\n",
      "  Node: 319, In-Degree Centrality: 5.771084337349397\n",
      "  Node: 370, In-Degree Centrality: 4.393574297188755\n",
      "  Node: 2778, In-Degree Centrality: 3.827309236947791\n",
      "  Node: 5149, In-Degree Centrality: 3.3775100401606424\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: GCCTGG, Out-Degree Centrality: 6.3212851405622486\n",
      "  Node: GGCTGG, Out-Degree Centrality: 5.899598393574297\n",
      "  Node: CTCAGC, Out-Degree Centrality: 4.923694779116466\n",
      "  Node: CTTTGG, Out-Degree Centrality: 3.919678714859437\n",
      "  Node: GCCCAG, Out-Degree Centrality: 3.811244979919678\n",
      "\n",
      "Community 39:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 2356, In-Degree Centrality: 19.764044943820224\n",
      "  Node: 1511, In-Degree Centrality: 13.202247191011235\n",
      "  Node: 10060, In-Degree Centrality: 13.0\n",
      "  Node: 5947, In-Degree Centrality: 11.719101123595506\n",
      "  Node: 4953, In-Degree Centrality: 11.573033707865168\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: TTGAAC, Out-Degree Centrality: 9.640449438202246\n",
      "  Node: CTTGAA, Out-Degree Centrality: 8.98876404494382\n",
      "  Node: GCTGGT, Out-Degree Centrality: 7.741573033707865\n",
      "  Node: CTCAAA, Out-Degree Centrality: 7.6966292134831455\n",
      "  Node: GCTTGA, Out-Degree Centrality: 7.280898876404494\n",
      "\n",
      "Community 40:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 3378, In-Degree Centrality: 65.46790890269152\n",
      "  Node: 5741, In-Degree Centrality: 8.157349896480332\n",
      "  Node: 314, In-Degree Centrality: 6.2815734989648035\n",
      "  Node: 6781, In-Degree Centrality: 5.701863354037267\n",
      "  Node: 4692, In-Degree Centrality: 4.15527950310559\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: CTCCTG, Out-Degree Centrality: 4.432712215320911\n",
      "  Node: CCTCCC, Out-Degree Centrality: 3.695652173913044\n",
      "  Node: TCCCAG, Out-Degree Centrality: 3.6935817805383024\n",
      "  Node: CTCCCA, Out-Degree Centrality: 3.298136645962733\n",
      "  Node: GCTCAC, Out-Degree Centrality: 2.428571428571429\n",
      "\n",
      "Community 41:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 3314, In-Degree Centrality: 26.506024096385545\n",
      "  Node: 1031, In-Degree Centrality: 12.987951807228916\n",
      "  Node: 8967, In-Degree Centrality: 12.55421686746988\n",
      "  Node: 6677, In-Degree Centrality: 10.240963855421688\n",
      "  Node: 3353, In-Degree Centrality: 9.650602409638555\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: GGAGGC, Out-Degree Centrality: 18.83132530120482\n",
      "  Node: TGTAAT, Out-Degree Centrality: 14.626506024096386\n",
      "  Node: TGCAGT, Out-Degree Centrality: 14.518072289156628\n",
      "  Node: TGAGGC, Out-Degree Centrality: 10.698795180722891\n",
      "  Node: AAAGTG, Out-Degree Centrality: 9.819277108433736\n",
      "\n",
      "Community 42:\n",
      " Central Features (In-Degree Centrality):\n",
      "  Node: 3794, In-Degree Centrality: 22.03076923076923\n",
      "  Node: 3582, In-Degree Centrality: 21.953846153846154\n",
      "  Node: 3476, In-Degree Centrality: 13.584615384615386\n",
      "  Node: 171, In-Degree Centrality: 13.476923076923077\n",
      "  Node: 3835, In-Degree Centrality: 12.230769230769232\n",
      " Central Motifs (Out-Degree Centrality):\n",
      "  Node: AGCCTC, Out-Degree Centrality: 15.446153846153846\n",
      "  Node: CTGGAG, Out-Degree Centrality: 14.307692307692308\n",
      "  Node: GCTGGA, Out-Degree Centrality: 13.53846153846154\n",
      "  Node: TGCTGG, Out-Degree Centrality: 12.723076923076924\n",
      "  Node: CCAAAG, Out-Degree Centrality: 12.430769230769231\n",
      "\n"
     ]
    }
   ],
   "source": [
    "subgraph_list = []\n",
    "for i, c in enumerate(communities):\n",
    "    graph = OG.subgraph(c)\n",
    "    central_features = sorted(nx.in_degree_centrality(graph).items(), key=lambda x: x[1], reverse=True)[:5]\n",
    "    central_motifs = sorted(nx.out_degree_centrality(graph).items(), key=lambda x: x[1], reverse=True)[:5]\n",
    "\n",
    "    print(f\"Community {i}:\")\n",
    "    print(\" Central Features (In-Degree Centrality):\")\n",
    "    for node, cent in central_features:\n",
    "        print(f\"  Node: {node}, In-Degree Centrality: {cent}\")\n",
    "    print(\" Central Motifs (Out-Degree Centrality):\")\n",
    "    for node, cent in central_motifs:\n",
    "        print(f\"  Node: {node}, Out-Degree Centrality: {cent}\")\n",
    "    graph = graph.to_undirected()\n",
    "\n",
    "    # Export each community to a separate JSON file\n",
    "    if not os.path.exists(f\"data/negKG_louvain_communities/{i}\"):\n",
    "        os.makedirs(f\"data/negKG_louvain_communities/{i}\", exist_ok=True)\n",
    "    with open(f\"data/negKG_louvain_communities/{i}/graph.json\", \"w\") as f:\n",
    "        json.dump(json_graph.node_link_data(graph, edges=\"edges\"), f)\n",
    "\n",
    "    subgraph_list.append({\"g\":graph, \"central_feat\": central_features[0], \"central_motif\": central_motifs[0]})\n",
    "    print(\"\")\n",
    "    # Visualize the community\n",
    "#    nx.draw(graph, pos=nx.kamada_kawai_layout(graph), with_labels=False, node_size=10, font_size=4, node_color=[node_color(n) for n in graph.nodes()])\n",
    "#    plt.title(f\"Community {i}\")\n",
    "#    plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "eb24c29b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# generate gene sets for central features and motifs in each community\n",
    "for i, sub in enumerate(subgraph_list):\n",
    "    graph = sub[\"g\"]\n",
    "    central_feat = sub[\"central_feat\"][0]\n",
    "    central_motif = sub[\"central_motif\"][0]\n",
    "\n",
    "    # central feature\n",
    "    if type(central_feat) == int:\n",
    "        edges = graph.edges(central_feat, data=\"annotations\")\n",
    "        genes = []\n",
    "        motifs = []\n",
    "        for u, v, annotations in edges:\n",
    "            for key, val in annotations.items():\n",
    "                motif = u if u != central_feat else v\n",
    "                genes.append(val[\"gene_id\"])\n",
    "                motifs.append(motif)\n",
    "        with open(f\"data/negKG_louvain_communities/{i}/feature_{central_feat}_genes.txt\", \"w\") as f:\n",
    "            f.writelines('\\n'.join(set(genes)))\n",
    "        with open(f\"data/negKG_louvain_communities/{i}/feature_{central_feat}_motifs.txt\", \"w\") as f:\n",
    "            f.writelines('\\n'.join(set(motifs)))\n",
    "\n",
    "    # central motif\n",
    "    if type(central_motif) == str:\n",
    "        edges = graph.edges(central_motif, data=\"annotations\")\n",
    "        genes = []\n",
    "        features = []\n",
    "        for u, v, annotations in edges:\n",
    "            for key, val in annotations.items():\n",
    "                feature = u if u != central_motif else v\n",
    "                genes.append(val[\"gene_id\"])\n",
    "                features.append(str(feature))\n",
    "        with open(f\"data/negKG_louvain_communities/{i}/motif_{central_motif}_genes.txt\", \"w\") as f:\n",
    "            f.writelines('\\n'.join(set(genes)))\n",
    "        with open(f\"data/negKG_louvain_communities/{i}/motif_{central_motif}_features.txt\", \"w\") as f:\n",
    "            f.writelines('\\n'.join(set(features)))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dd08b463",
   "metadata": {},
   "source": [
    "# Gene Set Enrichment Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "9d898601",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/neg_gene_sets/feature_8161_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/neg_gene_sets/enrich_8161\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/neg_gene_sets/feature_3378_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/neg_gene_sets/enrich_3378\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/neg_gene_sets/feature_7030_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/neg_gene_sets/enrich_7030\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, and Qing-Yu He. ChIPseeker: an\n",
      "R/Bioconductor package for ChIP peak annotation, comparison and\n",
      "visualization. Bioinformatics. 2015, 31(14):2382-2383\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/neg_gene_sets/all_coding_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/neg_gene_sets/enrich_coding\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/1/motif_AGAAAA_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/1/enrich_AGAAAA\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/1/feature_7030_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/1/enrich_7030\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/16/motif_GCCTCA_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error in file(file, ifelse(append, \"a\", \"w\")) : \n",
      "  cannot open the connection\n",
      "Calls: go_enrich ... write.csv -> eval.parent -> eval -> eval -> <Anonymous> -> file\n",
      "In addition: Warning message:\n",
      "In file(file, ifelse(append, \"a\", \"w\")) :\n",
      "  cannot open file 'data/negKG_louvain_communities/16/enrich_GCCTCA/go_enrichment_BP_results.csv': No such file or directory\n",
      "Execution halted\n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an\n",
      "R/Bioconductor package for Disease Ontology Semantic and Enrichment\n",
      "analysis. Bioinformatics. 2015, 31(4):608-609\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/16/feature_700_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error in file(file, ifelse(append, \"a\", \"w\")) : \n",
      "  cannot open the connection\n",
      "Calls: go_enrich ... write.csv -> eval.parent -> eval -> eval -> <Anonymous> -> file\n",
      "In addition: Warning message:\n",
      "In file(file, ifelse(append, \"a\", \"w\")) :\n",
      "  cannot open file 'data/negKG_louvain_communities/16/enrich_700/go_enrichment_BP_results.csv': No such file or directory\n",
      "Execution halted\n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an\n",
      "R/Bioconductor package for Disease Ontology Semantic and Enrichment\n",
      "analysis. Bioinformatics. 2015, 31(4):608-609\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/33/motif_CACTTT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error in file(file, ifelse(append, \"a\", \"w\")) : \n",
      "  cannot open the connection\n",
      "Calls: go_enrich ... write.csv -> eval.parent -> eval -> eval -> <Anonymous> -> file\n",
      "In addition: Warning message:\n",
      "In file(file, ifelse(append, \"a\", \"w\")) :\n",
      "  cannot open file 'data/negKG_louvain_communities/33/enrich_CACTTT/go_enrichment_BP_results.csv': No such file or directory\n",
      "Execution halted\n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an\n",
      "R/Bioconductor package for Disease Ontology Semantic and Enrichment\n",
      "analysis. Bioinformatics. 2015, 31(4):608-609\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/33/feature_8097_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/33/enrich_8097\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an\n",
      "R/Bioconductor package for Disease Ontology Semantic and Enrichment\n",
      "analysis. Bioinformatics. 2015, 31(4):608-609\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/42/feature_3794_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/42/enrich_3794\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/42/motif_AGCCTC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/42/enrich_AGCCTC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an\n",
      "R/Bioconductor package for Disease Ontology Semantic and Enrichment\n",
      "analysis. Bioinformatics. 2015, 31(4):608-609\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/13/motif_CAGCCT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/13/enrich_CAGCCT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/13/feature_1261_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/13/enrich_1261\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an\n",
      "R/Bioconductor package for Disease Ontology Semantic and Enrichment\n",
      "analysis. Bioinformatics. 2015, 31(4):608-609\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/14/motif_GCCTGT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/14/enrich_GCCTGT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/14/feature_6912_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/14/enrich_6912\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/28/feature_4955_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/28/enrich_4955\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/28/motif_AGTGCT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/28/enrich_AGTGCT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/21/feature_7664_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/21/enrich_7664\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/21/motif_CACCTG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error in file(file, ifelse(append, \"a\", \"w\")) : \n",
      "  cannot open the connection\n",
      "Calls: go_enrich ... write.csv -> eval.parent -> eval -> eval -> <Anonymous> -> file\n",
      "In addition: Warning message:\n",
      "In file(file, ifelse(append, \"a\", \"w\")) :\n",
      "  cannot open file 'data/negKG_louvain_communities/21/enrich_CACCTG/go_enrichment_BP_results.csv': No such file or directory\n",
      "Execution halted\n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/3/motif_GGGATT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/3/enrich_GGGATT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/3/feature_1389_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/3/enrich_1389\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/30/feature_5002_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/30/enrich_5002\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Using meshes for MeSH term enrichment and semantic\n",
      "analyses. Bioinformatics. 2018, 34(21):3766-3767,\n",
      "doi:10.1093/bioinformatics/bty410\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/30/motif_GGTTCA_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/30/enrich_GGTTCA\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/12/motif_CACTGC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/12/enrich_CACTGC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/12/feature_8356_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/12/enrich_8356\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/15/feature_8161_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/15/enrich_8161\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/15/motif_ACATTT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/15/enrich_ACATTT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/39/motif_TTGAAC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/39/enrich_TTGAAC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/39/feature_2356_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/39/enrich_2356\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/27/motif_CAGGAG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/27/enrich_CAGGAG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/27/feature_8911_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/27/enrich_8911\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/2/feature_1845_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/2/enrich_1845\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/2/motif_G_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/2/enrich_G\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/9/feature_8453_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/9/enrich_8453\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/9/motif_CTGGGA_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/9/enrich_CTGGGA\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/34/feature_3384_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/34/enrich_3384\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/34/motif_GGCTCA_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error in file(file, ifelse(append, \"a\", \"w\")) : \n",
      "  cannot open the connection\n",
      "Calls: go_enrich ... write.csv -> eval.parent -> eval -> eval -> <Anonymous> -> file\n",
      "In addition: Warning message:\n",
      "In file(file, ifelse(append, \"a\", \"w\")) :\n",
      "  cannot open file 'data/negKG_louvain_communities/34/enrich_GGCTCA/go_enrichment_BP_results.csv': No such file or directory\n",
      "Execution halted\n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/32/motif_CCAGCT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/32/enrich_CCAGCT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/32/feature_7320_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error in file(file, ifelse(append, \"a\", \"w\")) : \n",
      "  cannot open the connection\n",
      "Calls: go_enrich ... write.csv -> eval.parent -> eval -> eval -> <Anonymous> -> file\n",
      "In addition: Warning message:\n",
      "In file(file, ifelse(append, \"a\", \"w\")) :\n",
      "  cannot open file 'data/negKG_louvain_communities/32/enrich_7320/go_enrichment_BP_results.csv': No such file or directory\n",
      "Execution halted\n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/18/motif_TGCCTC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/18/enrich_TGCCTC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Using meshes for MeSH term enrichment and semantic\n",
      "analyses. Bioinformatics. 2018, 34(21):3766-3767,\n",
      "doi:10.1093/bioinformatics/bty410\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/18/feature_329_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/18/enrich_329\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/19/motif_GCAGTG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/19/enrich_GCAGTG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/19/feature_7622_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error in file(file, ifelse(append, \"a\", \"w\")) : \n",
      "  cannot open the connection\n",
      "Calls: go_enrich ... write.csv -> eval.parent -> eval -> eval -> <Anonymous> -> file\n",
      "In addition: Warning message:\n",
      "In file(file, ifelse(append, \"a\", \"w\")) :\n",
      "  cannot open file 'data/negKG_louvain_communities/19/enrich_7622/go_enrichment_BP_results.csv': No such file or directory\n",
      "Execution halted\n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/7/motif_TGGGAT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/7/enrich_TGGGAT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/7/feature_1636_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/7/enrich_1636\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/23/motif_CCAGGC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/23/enrich_CCAGGC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/23/feature_7714_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/23/enrich_7714\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Using meshes for MeSH term enrichment and semantic\n",
      "analyses. Bioinformatics. 2018, 34(21):3766-3767,\n",
      "doi:10.1093/bioinformatics/bty410\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/17/motif_GCTGGG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/17/enrich_GCTGGG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Using meshes for MeSH term enrichment and semantic\n",
      "analyses. Bioinformatics. 2018, 34(21):3766-3767,\n",
      "doi:10.1093/bioinformatics/bty410\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/17/feature_1052_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/17/enrich_1052\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Using meshes for MeSH term enrichment and semantic\n",
      "analyses. Bioinformatics. 2018, 34(21):3766-3767,\n",
      "doi:10.1093/bioinformatics/bty410\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/29/feature_7222_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/29/enrich_7222\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/29/motif_GAGTGC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/29/enrich_GAGTGC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Using meshes for MeSH term enrichment and semantic\n",
      "analyses. Bioinformatics. 2018, 34(21):3766-3767,\n",
      "doi:10.1093/bioinformatics/bty410\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/11/feature_8853_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/11/enrich_8853\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/11/motif_AATTAG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error in file(file, ifelse(append, \"a\", \"w\")) : \n",
      "  cannot open the connection\n",
      "Calls: go_enrich ... write.csv -> eval.parent -> eval -> eval -> <Anonymous> -> file\n",
      "In addition: Warning message:\n",
      "In file(file, ifelse(append, \"a\", \"w\")) :\n",
      "  cannot open file 'data/negKG_louvain_communities/11/enrich_AATTAG/go_enrichment_BP_results.csv': No such file or directory\n",
      "Execution halted\n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/41/motif_GGAGGC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error in file(file, ifelse(append, \"a\", \"w\")) : \n",
      "  cannot open the connection\n",
      "Calls: go_enrich ... write.csv -> eval.parent -> eval -> eval -> <Anonymous> -> file\n",
      "In addition: Warning message:\n",
      "In file(file, ifelse(append, \"a\", \"w\")) :\n",
      "  cannot open file 'data/negKG_louvain_communities/41/enrich_GGAGGC/go_enrichment_BP_results.csv': No such file or directory\n",
      "Execution halted\n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/41/feature_3314_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/41/enrich_3314\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/26/feature_9291_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/26/enrich_9291\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an\n",
      "R/Bioconductor package for Disease Ontology Semantic and Enrichment\n",
      "analysis. Bioinformatics. 2015, 31(4):608-609\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/26/motif_TTACAG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/26/enrich_TTACAG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/8/feature_6445_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/8/enrich_6445\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, and Qing-Yu He. ChIPseeker: an\n",
      "R/Bioconductor package for ChIP peak annotation, comparison and\n",
      "visualization. Bioinformatics. 2015, 31(14):2382-2383\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/8/motif_GGGAGG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/8/enrich_GGGAGG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/40/motif_CTCCTG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/40/enrich_CTCCTG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, and Qing-Yu He. ChIPseeker: an\n",
      "R/Bioconductor package for ChIP peak annotation, comparison and\n",
      "visualization. Bioinformatics. 2015, 31(14):2382-2383\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/40/feature_3378_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/40/enrich_3378\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/10/motif_AGGCTG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/10/enrich_AGGCTG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, and Qing-Yu He. ChIPseeker: an\n",
      "R/Bioconductor package for ChIP peak annotation, comparison and\n",
      "visualization. Bioinformatics. 2015, 31(14):2382-2383\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/10/feature_8358_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/10/enrich_8358\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/5/feature_4793_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/5/enrich_4793\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/5/motif_TGTGTG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/5/enrich_TGTGTG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/25/motif_TACAGG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/25/enrich_TACAGG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/25/feature_5871_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/25/enrich_5871\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/6/motif_TTTTTT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/6/enrich_TTTTTT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/6/feature_4831_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/6/enrich_4831\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/4/motif_AAAAAA_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/4/enrich_AAAAAA\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/4/feature_4814_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/4/enrich_4814\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/31/feature_6254_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/31/enrich_6254\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/31/motif_CCTGCC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error in file(file, ifelse(append, \"a\", \"w\")) : \n",
      "  cannot open the connection\n",
      "Calls: go_enrich ... write.csv -> eval.parent -> eval -> eval -> <Anonymous> -> file\n",
      "In addition: Warning message:\n",
      "In file(file, ifelse(append, \"a\", \"w\")) :\n",
      "  cannot open file 'data/negKG_louvain_communities/31/enrich_CCTGCC/go_enrichment_BP_results.csv': No such file or directory\n",
      "Execution halted\n",
      "sh: 1: Syntax error: end of file unexpected\n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/0/feature_5746_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/0/enrich_5746\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/37/motif_AGTGCA_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/37/enrich_AGTGCA\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, and Qing-Yu He. ChIPseeker: an\n",
      "R/Bioconductor package for ChIP peak annotation, comparison and\n",
      "visualization. Bioinformatics. 2015, 31(14):2382-2383\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/37/feature_3568_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/37/enrich_3568\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, and Qing-Yu He. ChIPseeker: an\n",
      "R/Bioconductor package for ChIP peak annotation, comparison and\n",
      "visualization. Bioinformatics. 2015, 31(14):2382-2383\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/35/motif_AGCCTG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/35/enrich_AGCCTG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/35/feature_4494_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/35/enrich_4494\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/20/motif_CCCAGC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/20/enrich_CCCAGC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/20/feature_7990_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/20/enrich_7990\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/38/feature_4014_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/38/enrich_4014\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Using meshes for MeSH term enrichment and semantic\n",
      "analyses. Bioinformatics. 2018, 34(21):3766-3767,\n",
      "doi:10.1093/bioinformatics/bty410\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/38/motif_GCCTGG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/38/enrich_GCCTGG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/36/motif_TTGTAT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error in file(file, ifelse(append, \"a\", \"w\")) : \n",
      "  cannot open the connection\n",
      "Calls: go_enrich ... write.csv -> eval.parent -> eval -> eval -> <Anonymous> -> file\n",
      "In addition: Warning message:\n",
      "In file(file, ifelse(append, \"a\", \"w\")) :\n",
      "  cannot open file 'data/negKG_louvain_communities/36/enrich_TTGTAT/go_enrichment_BP_results.csv': No such file or directory\n",
      "Execution halted\n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/36/feature_8204_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/36/enrich_8204\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/22/feature_10234_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/22/enrich_10234\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/22/motif_GGTCTC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/22/enrich_GGTCTC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/24/feature_526_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/24/enrich_526\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/24/motif_GCCTCC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: BP\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/24/enrich_GCCTCC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n"
     ]
    }
   ],
   "source": [
    "# run R enrichment analysis on these gene sets using our custom script\n",
    "\n",
    "neg_gene_sets = [f for f in os.listdir(\"data/neg_gene_sets\") if f.endswith(\"_genes.txt\")]\n",
    "for gs in neg_gene_sets:\n",
    "    os.system(f\"Rscript enrich.R --gene_list data/neg_gene_sets/{gs} --outdir data/neg_gene_sets/enrich_{gs.split('_')[1]}\")\n",
    "\n",
    "for c in os.listdir(\"data/negKG_louvain_communities\"):\n",
    "    neg_gene_sets = [f for f in os.listdir(f\"data/negKG_louvain_communities/{c}\") if f.endswith(\"_genes.txt\")]\n",
    "    for gs in neg_gene_sets:\n",
    "        os.system(f\"Rscript enrich.R --gene_list data/negKG_louvain_communities/{c}/{gs} --outdir data/negKG_louvain_communities/{c}/enrich_{gs.split('_')[1]}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "a4c4bdc7",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an\n",
      "R/Bioconductor package for Disease Ontology Semantic and Enrichment\n",
      "analysis. Bioinformatics. 2015, 31(4):608-609\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/neg_gene_sets/feature_8161_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/neg_gene_sets/enrich_8161\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/neg_gene_sets/feature_3378_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/neg_gene_sets/enrich_3378\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/neg_gene_sets/feature_7030_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/neg_gene_sets/enrich_7030\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/neg_gene_sets/all_coding_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/neg_gene_sets/enrich_coding\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/1/motif_AGAAAA_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/1/enrich_AGAAAA\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/1/feature_7030_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/1/enrich_7030\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/16/motif_GCCTCA_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/16/enrich_GCCTCA\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/16/feature_700_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/16/enrich_700\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Using meshes for MeSH term enrichment and semantic\n",
      "analyses. Bioinformatics. 2018, 34(21):3766-3767,\n",
      "doi:10.1093/bioinformatics/bty410\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/33/motif_CACTTT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/33/enrich_CACTTT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/33/feature_8097_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/33/enrich_8097\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/42/feature_3794_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Using meshes for MeSH term enrichment and semantic\n",
      "analyses. Bioinformatics. 2018, 34(21):3766-3767,\n",
      "doi:10.1093/bioinformatics/bty410\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/42/motif_AGCCTC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/42/enrich_AGCCTC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/13/motif_CAGCCT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/13/enrich_CAGCCT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/13/feature_1261_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/13/enrich_1261\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/14/motif_GCCTGT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/14/feature_6912_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/14/enrich_6912\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, and Qing-Yu He. ChIPseeker: an\n",
      "R/Bioconductor package for ChIP peak annotation, comparison and\n",
      "visualization. Bioinformatics. 2015, 31(14):2382-2383\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/28/feature_4955_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/28/enrich_4955\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/28/motif_AGTGCT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/28/enrich_AGTGCT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/21/feature_7664_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/21/enrich_7664\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/21/motif_CACCTG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error in file(file, ifelse(append, \"a\", \"w\")) : \n",
      "  cannot open the connection\n",
      "Calls: go_enrich ... write.csv -> eval.parent -> eval -> eval -> <Anonymous> -> file\n",
      "In addition: Warning message:\n",
      "In file(file, ifelse(append, \"a\", \"w\")) :\n",
      "  cannot open file 'data/negKG_louvain_communities/21/enrich_CACCTG/go_enrichment_MF_results.csv': No such file or directory\n",
      "Execution halted\n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/3/motif_GGGATT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, and Qing-Yu He. ChIPseeker: an\n",
      "R/Bioconductor package for ChIP peak annotation, comparison and\n",
      "visualization. Bioinformatics. 2015, 31(14):2382-2383\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/3/feature_1389_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/3/enrich_1389\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/30/feature_5002_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/30/enrich_5002\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/30/motif_GGTTCA_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/30/enrich_GGTTCA\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/12/motif_CACTGC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/12/enrich_CACTGC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an\n",
      "R/Bioconductor package for Disease Ontology Semantic and Enrichment\n",
      "analysis. Bioinformatics. 2015, 31(4):608-609\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/12/feature_8356_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/12/enrich_8356\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/15/feature_8161_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/15/enrich_8161\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/15/motif_ACATTT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/15/enrich_ACATTT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/39/motif_TTGAAC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/39/enrich_TTGAAC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/39/feature_2356_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/39/enrich_2356\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/27/motif_CAGGAG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/27/enrich_CAGGAG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Using meshes for MeSH term enrichment and semantic\n",
      "analyses. Bioinformatics. 2018, 34(21):3766-3767,\n",
      "doi:10.1093/bioinformatics/bty410\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/27/feature_8911_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/27/enrich_8911\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/2/feature_1845_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/2/enrich_1845\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/2/motif_G_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/2/enrich_G\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/9/feature_8453_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/9/enrich_8453\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/9/motif_CTGGGA_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/9/enrich_CTGGGA\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/34/feature_3384_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/34/enrich_3384\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, and Qing-Yu He. ChIPseeker: an\n",
      "R/Bioconductor package for ChIP peak annotation, comparison and\n",
      "visualization. Bioinformatics. 2015, 31(14):2382-2383\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/34/motif_GGCTCA_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error in file(file, ifelse(append, \"a\", \"w\")) : \n",
      "  cannot open the connection\n",
      "Calls: go_enrich ... write.csv -> eval.parent -> eval -> eval -> <Anonymous> -> file\n",
      "In addition: Warning message:\n",
      "In file(file, ifelse(append, \"a\", \"w\")) :\n",
      "  cannot open file 'data/negKG_louvain_communities/34/enrich_GGCTCA/go_enrichment_MF_results.csv': No such file or directory\n",
      "Execution halted\n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an\n",
      "R/Bioconductor package for Disease Ontology Semantic and Enrichment\n",
      "analysis. Bioinformatics. 2015, 31(4):608-609\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/32/motif_CCAGCT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/32/enrich_CCAGCT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an\n",
      "R/Bioconductor package for Disease Ontology Semantic and Enrichment\n",
      "analysis. Bioinformatics. 2015, 31(4):608-609\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/32/feature_7320_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error in file(file, ifelse(append, \"a\", \"w\")) : \n",
      "  cannot open the connection\n",
      "Calls: go_enrich ... write.csv -> eval.parent -> eval -> eval -> <Anonymous> -> file\n",
      "In addition: Warning message:\n",
      "In file(file, ifelse(append, \"a\", \"w\")) :\n",
      "  cannot open file 'data/negKG_louvain_communities/32/enrich_7320/go_enrichment_MF_results.csv': No such file or directory\n",
      "Execution halted\n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/18/motif_TGCCTC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/18/feature_329_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/18/enrich_329\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/19/motif_GCAGTG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/19/enrich_GCAGTG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/19/feature_7622_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/19/enrich_7622\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/7/motif_TGGGAT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/7/enrich_TGGGAT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, and Qing-Yu He. ChIPseeker: an\n",
      "R/Bioconductor package for ChIP peak annotation, comparison and\n",
      "visualization. Bioinformatics. 2015, 31(14):2382-2383\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/7/feature_1636_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/7/enrich_1636\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/23/motif_CCAGGC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/23/enrich_CCAGGC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/23/feature_7714_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/23/enrich_7714\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/17/motif_GCTGGG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/17/enrich_GCTGGG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/17/feature_1052_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/17/enrich_1052\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/29/feature_7222_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/29/motif_GAGTGC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/11/feature_8853_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/11/enrich_8853\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/11/motif_AATTAG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error in file(file, ifelse(append, \"a\", \"w\")) : \n",
      "  cannot open the connection\n",
      "Calls: go_enrich ... write.csv -> eval.parent -> eval -> eval -> <Anonymous> -> file\n",
      "In addition: Warning message:\n",
      "In file(file, ifelse(append, \"a\", \"w\")) :\n",
      "  cannot open file 'data/negKG_louvain_communities/11/enrich_AATTAG/go_enrichment_MF_results.csv': No such file or directory\n",
      "Execution halted\n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/41/motif_GGAGGC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error in file(file, ifelse(append, \"a\", \"w\")) : \n",
      "  cannot open the connection\n",
      "Calls: go_enrich ... write.csv -> eval.parent -> eval -> eval -> <Anonymous> -> file\n",
      "In addition: Warning message:\n",
      "In file(file, ifelse(append, \"a\", \"w\")) :\n",
      "  cannot open file 'data/negKG_louvain_communities/41/enrich_GGAGGC/go_enrichment_MF_results.csv': No such file or directory\n",
      "Execution halted\n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an\n",
      "R/Bioconductor package for Disease Ontology Semantic and Enrichment\n",
      "analysis. Bioinformatics. 2015, 31(4):608-609\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/41/feature_3314_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/41/enrich_3314\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Using meshes for MeSH term enrichment and semantic\n",
      "analyses. Bioinformatics. 2018, 34(21):3766-3767,\n",
      "doi:10.1093/bioinformatics/bty410\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/26/feature_9291_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/26/enrich_9291\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/26/motif_TTACAG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/26/enrich_TTACAG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/8/feature_6445_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/8/enrich_6445\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/8/motif_GGGAGG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/8/enrich_GGGAGG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/40/motif_CTCCTG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/40/enrich_CTCCTG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/40/feature_3378_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/40/enrich_3378\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, and Qing-Yu He. ChIPseeker: an\n",
      "R/Bioconductor package for ChIP peak annotation, comparison and\n",
      "visualization. Bioinformatics. 2015, 31(14):2382-2383\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/10/motif_AGGCTG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/10/enrich_AGGCTG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/10/feature_8358_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/10/enrich_8358\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/5/feature_4793_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/5/motif_TGTGTG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/5/enrich_TGTGTG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/25/motif_TACAGG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/25/feature_5871_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/25/enrich_5871\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/6/motif_TTTTTT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/6/enrich_TTTTTT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/6/feature_4831_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/6/enrich_4831\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/4/motif_AAAAAA_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/4/enrich_AAAAAA\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, and Qing-Yu He. ChIPseeker: an\n",
      "R/Bioconductor package for ChIP peak annotation, comparison and\n",
      "visualization. Bioinformatics. 2015, 31(14):2382-2383\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/4/feature_4814_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/4/enrich_4814\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/31/feature_6254_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/31/enrich_6254\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/31/motif_CCTGCC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error in file(file, ifelse(append, \"a\", \"w\")) : \n",
      "  cannot open the connection\n",
      "Calls: go_enrich ... write.csv -> eval.parent -> eval -> eval -> <Anonymous> -> file\n",
      "In addition: Warning message:\n",
      "In file(file, ifelse(append, \"a\", \"w\")) :\n",
      "  cannot open file 'data/negKG_louvain_communities/31/enrich_CCTGCC/go_enrichment_MF_results.csv': No such file or directory\n",
      "Execution halted\n",
      "sh: 1: cannot open cls: No such file\n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/0/feature_5746_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/0/enrich_5746\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/37/motif_AGTGCA_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/37/enrich_AGTGCA\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/37/feature_3568_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/37/enrich_3568\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/35/motif_AGCCTG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/35/enrich_AGCCTG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/35/feature_4494_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, and Qing-Yu He. ChIPseeker: an\n",
      "R/Bioconductor package for ChIP peak annotation, comparison and\n",
      "visualization. Bioinformatics. 2015, 31(14):2382-2383\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/20/motif_CCCAGC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/20/enrich_CCCAGC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/20/feature_7990_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/38/feature_4014_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/38/enrich_4014\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/38/motif_GCCTGG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/38/enrich_GCCTGG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/36/motif_TTGTAT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/36/enrich_TTGTAT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/36/feature_8204_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/22/feature_10234_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/22/enrich_10234\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/22/motif_GGTCTC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/24/feature_526_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/24/enrich_526\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/24/motif_GCCTCC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: MF\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/24/enrich_GCCTCC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n"
     ]
    }
   ],
   "source": [
    "# molecular function enrichment as well\n",
    "\n",
    "neg_gene_sets = [f for f in os.listdir(\"data/neg_gene_sets\") if f.endswith(\"_genes.txt\")]\n",
    "for gs in neg_gene_sets:\n",
    "    os.system(f\"Rscript enrich.R --gene_list data/neg_gene_sets/{gs} --outdir data/neg_gene_sets/enrich_{gs.split('_')[1]} --ont MF\")\n",
    "\n",
    "for c in os.listdir(\"data/negKG_louvain_communities\"):\n",
    "    neg_gene_sets = [f for f in os.listdir(f\"data/negKG_louvain_communities/{c}\") if f.endswith(\"_genes.txt\")]\n",
    "    for gs in neg_gene_sets:\n",
    "        os.system(f\"Rscript enrich.R --gene_list data/negKG_louvain_communities/{c}/{gs} --outdir data/negKG_louvain_communities/{c}/enrich_{gs.split('_')[1]} --ont MF\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "fb2b1af5",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/neg_gene_sets/feature_8161_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/neg_gene_sets/enrich_8161\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/neg_gene_sets/feature_3378_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/neg_gene_sets/enrich_3378\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, and Qing-Yu He. ChIPseeker: an\n",
      "R/Bioconductor package for ChIP peak annotation, comparison and\n",
      "visualization. Bioinformatics. 2015, 31(14):2382-2383\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/neg_gene_sets/feature_7030_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/neg_gene_sets/enrich_7030\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Using meshes for MeSH term enrichment and semantic\n",
      "analyses. Bioinformatics. 2018, 34(21):3766-3767,\n",
      "doi:10.1093/bioinformatics/bty410\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/neg_gene_sets/all_coding_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/neg_gene_sets/enrich_coding\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/1/motif_AGAAAA_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/1/enrich_AGAAAA\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/1/feature_7030_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/1/enrich_7030\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/16/motif_GCCTCA_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/16/enrich_GCCTCA\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/16/feature_700_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/16/enrich_700\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/33/motif_CACTTT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/33/enrich_CACTTT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/33/feature_8097_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/33/enrich_8097\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/42/feature_3794_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/42/enrich_3794\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/42/motif_AGCCTC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/42/enrich_AGCCTC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an\n",
      "R/Bioconductor package for Disease Ontology Semantic and Enrichment\n",
      "analysis. Bioinformatics. 2015, 31(4):608-609\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/13/motif_CAGCCT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/13/enrich_CAGCCT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/13/feature_1261_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/13/enrich_1261\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/14/motif_GCCTGT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/14/enrich_GCCTGT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/14/feature_6912_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/14/enrich_6912\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/28/feature_4955_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/28/enrich_4955\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/28/motif_AGTGCT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/28/enrich_AGTGCT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/21/feature_7664_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/21/enrich_7664\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/21/motif_CACCTG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/21/enrich_CACCTG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/3/motif_GGGATT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/3/enrich_GGGATT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Using meshes for MeSH term enrichment and semantic\n",
      "analyses. Bioinformatics. 2018, 34(21):3766-3767,\n",
      "doi:10.1093/bioinformatics/bty410\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/3/feature_1389_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/3/enrich_1389\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Using meshes for MeSH term enrichment and semantic\n",
      "analyses. Bioinformatics. 2018, 34(21):3766-3767,\n",
      "doi:10.1093/bioinformatics/bty410\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/30/feature_5002_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/30/enrich_5002\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/30/motif_GGTTCA_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/30/enrich_GGTTCA\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/12/motif_CACTGC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/12/enrich_CACTGC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Using meshes for MeSH term enrichment and semantic\n",
      "analyses. Bioinformatics. 2018, 34(21):3766-3767,\n",
      "doi:10.1093/bioinformatics/bty410\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/12/feature_8356_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/12/enrich_8356\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/15/feature_8161_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/15/enrich_8161\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/15/motif_ACATTT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/15/enrich_ACATTT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an\n",
      "R/Bioconductor package for Disease Ontology Semantic and Enrichment\n",
      "analysis. Bioinformatics. 2015, 31(4):608-609\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/39/motif_TTGAAC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/39/enrich_TTGAAC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/39/feature_2356_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/39/enrich_2356\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/27/motif_CAGGAG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/27/enrich_CAGGAG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/27/feature_8911_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/27/enrich_8911\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Using meshes for MeSH term enrichment and semantic\n",
      "analyses. Bioinformatics. 2018, 34(21):3766-3767,\n",
      "doi:10.1093/bioinformatics/bty410\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/2/feature_1845_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/2/enrich_1845\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/2/motif_G_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/2/enrich_G\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/9/feature_8453_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/9/enrich_8453\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/9/motif_CTGGGA_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/9/enrich_CTGGGA\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/34/feature_3384_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/34/enrich_3384\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Using meshes for MeSH term enrichment and semantic\n",
      "analyses. Bioinformatics. 2018, 34(21):3766-3767,\n",
      "doi:10.1093/bioinformatics/bty410\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/34/motif_GGCTCA_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Error in file(file, ifelse(append, \"a\", \"w\")) : \n",
      "  cannot open the connection\n",
      "Calls: go_enrich ... write.csv -> eval.parent -> eval -> eval -> <Anonymous> -> file\n",
      "In addition: Warning message:\n",
      "In file(file, ifelse(append, \"a\", \"w\")) :\n",
      "  cannot open file 'data/negKG_louvain_communities/34/enrich_GGCTCA/go_enrichment_CC_results.csv': No such file or directory\n",
      "Execution halted\n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/32/motif_CCAGCT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/32/enrich_CCAGCT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/32/feature_7320_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/32/enrich_7320\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/18/motif_TGCCTC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/18/enrich_TGCCTC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Using meshes for MeSH term enrichment and semantic\n",
      "analyses. Bioinformatics. 2018, 34(21):3766-3767,\n",
      "doi:10.1093/bioinformatics/bty410\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/18/feature_329_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/18/enrich_329\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/19/motif_GCAGTG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/19/enrich_GCAGTG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/19/feature_7622_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/19/enrich_7622\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/7/motif_TGGGAT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/7/enrich_TGGGAT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/7/feature_1636_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/7/enrich_1636\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/23/motif_CCAGGC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/23/enrich_CCAGGC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/23/feature_7714_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/23/enrich_7714\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/17/motif_GCTGGG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/17/enrich_GCTGGG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/17/feature_1052_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/17/enrich_1052\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/29/feature_7222_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/29/enrich_7222\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/29/motif_GAGTGC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/29/enrich_GAGTGC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/11/feature_8853_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/11/enrich_8853\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/11/motif_AATTAG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/11/enrich_AATTAG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, and Qing-Yu He. ChIPseeker: an\n",
      "R/Bioconductor package for ChIP peak annotation, comparison and\n",
      "visualization. Bioinformatics. 2015, 31(14):2382-2383\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/41/motif_GGAGGC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/41/enrich_GGAGGC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/41/feature_3314_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/41/enrich_3314\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/26/feature_9291_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/26/enrich_9291\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/26/motif_TTACAG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/26/enrich_TTACAG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/8/feature_6445_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/8/enrich_6445\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/8/motif_GGGAGG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/8/enrich_GGGAGG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/40/motif_CTCCTG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/40/enrich_CTCCTG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Using meshes for MeSH term enrichment and semantic\n",
      "analyses. Bioinformatics. 2018, 34(21):3766-3767,\n",
      "doi:10.1093/bioinformatics/bty410\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/40/feature_3378_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/40/enrich_3378\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/10/motif_AGGCTG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/10/enrich_AGGCTG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/10/feature_8358_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/10/enrich_8358\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, and Qing-Yu He. ChIPseeker: an\n",
      "R/Bioconductor package for ChIP peak annotation, comparison and\n",
      "visualization. Bioinformatics. 2015, 31(14):2382-2383\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/5/feature_4793_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/5/enrich_4793\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/5/motif_TGTGTG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/5/enrich_TGTGTG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/25/motif_TACAGG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/25/enrich_TACAGG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu and Shengqi\n",
      "Wang. GOSemSim: an R package for measuring semantic similarity among GO\n",
      "terms and gene products. Bioinformatics. 2010, 26(7):976-978\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/25/feature_5871_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/25/enrich_5871\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/6/motif_TTTTTT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/6/enrich_TTTTTT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Using meshes for MeSH term enrichment and semantic\n",
      "analyses. Bioinformatics. 2018, 34(21):3766-3767,\n",
      "doi:10.1093/bioinformatics/bty410\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/6/feature_4831_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/6/enrich_4831\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/4/motif_AAAAAA_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/4/enrich_AAAAAA\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/4/feature_4814_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/4/enrich_4814\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/31/feature_6254_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/31/enrich_6254\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/31/motif_CCTGCC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/31/enrich_CCTGCC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "sh: 1: cannot open cls: No such file\n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an\n",
      "R/Bioconductor package for Disease Ontology Semantic and Enrichment\n",
      "analysis. Bioinformatics. 2015, 31(4):608-609\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/0/feature_5746_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/0/enrich_5746\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Qing-Yu He. ReactomePA: an R/Bioconductor package for\n",
      "reactome pathway analysis and visualization. Molecular BioSystems.\n",
      "2016, 12(2):477-479\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/37/motif_AGTGCA_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/37/enrich_AGTGCA\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/37/feature_3568_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/37/enrich_3568\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an\n",
      "R/Bioconductor package for Disease Ontology Semantic and Enrichment\n",
      "analysis. Bioinformatics. 2015, 31(4):608-609\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/35/motif_AGCCTG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/35/enrich_AGCCTG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an\n",
      "R/Bioconductor package for Disease Ontology Semantic and Enrichment\n",
      "analysis. Bioinformatics. 2015, 31(4):608-609\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/35/feature_4494_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/35/enrich_4494\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/20/motif_CCCAGC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/20/enrich_CCCAGC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/20/feature_7990_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/20/enrich_7990\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, and Qing-Yu He. ChIPseeker: an\n",
      "R/Bioconductor package for ChIP peak annotation, comparison and\n",
      "visualization. Bioinformatics. 2015, 31(14):2382-2383\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/38/feature_4014_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/38/enrich_4014\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/38/motif_GCCTGG_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/38/enrich_GCCTGG\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/36/motif_TTGTAT_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/36/enrich_TTGTAT\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "S Xu, E Hu, Y Cai, Z Xie, X Luo, L Zhan, W Tang, Q Wang, B Liu, R Wang,\n",
      "W Xie, T Wu, L Xie, G Yu. Using clusterProfiler to characterize\n",
      "multiomics data. Nature Protocols. 2024, 19(11):3292-3320\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using\n",
      "GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks.\n",
      "Methods in Molecular Biology. 2020, 2117:207-215. Humana, New York, NY.\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/36/feature_8204_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/36/enrich_8204\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/22/feature_10234_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/22/enrich_10234\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "G Yu. Thirteen years of clusterProfiler. The Innovation. 2024,\n",
      "5(6):100722\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Using meshes for MeSH term enrichment and semantic\n",
      "analyses. Bioinformatics. 2018, 34(21):3766-3767,\n",
      "doi:10.1093/bioinformatics/bty410\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/22/motif_GGTCTC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"No significant enrichment results found.\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu, Li-Gen Wang, Yanyan Han and Qing-Yu He.\n",
      "clusterProfiler: an R package for comparing biological themes among\n",
      "gene clusters. OMICS: A Journal of Integrative Biology. 2012,\n",
      "16(5):284-287\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Guangchuang Yu. Using meshes for MeSH term enrichment and semantic\n",
      "analyses. Bioinformatics. 2018, 34(21):3766-3767,\n",
      "doi:10.1093/bioinformatics/bty410\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/24/feature_526_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/24/enrich_526\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n",
      "\n",
      "clusterProfiler v4.16.0 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "T Wu, E Hu, S Xu, M Chen, P Guo, Z Dai, T Feng, L Zhou, W Tang, L Zhan,\n",
      "X Fu, S Liu, X Bo, and G Yu. clusterProfiler 4.0: A universal\n",
      "enrichment tool for interpreting omics data. The Innovation. 2021,\n",
      "2(3):100141\n",
      "\n",
      "Attaching package: clusterProfiler\n",
      "\n",
      "The following object is masked from package:stats:\n",
      "\n",
      "    filter\n",
      "\n",
      "enrichplot v1.28.4 Learn more at https://yulab-smu.top/contribution-knowledge-mining/\n",
      "\n",
      "Please cite:\n",
      "\n",
      "Qianwen Wang, Ming Li, Tianzhi Wu, Li Zhan, Lin Li, Meijun Chen, Wenqin\n",
      "Xie, Zijing Xie, Erqiang Hu, Shuangbin Xu, Guangchuang Yu. Exploring\n",
      "epigenomic datasets by ChIPseeker. Current Protocols. 2022, 2(10): e585\n",
      "Welcome to enrichR\n",
      "Checking connections ... \n",
      "Enrichr ... Connection is Live!\n",
      "FlyEnrichr ... Connection is Live!\n",
      "WormEnrichr ... Connection is Live!\n",
      "YeastEnrichr ... Connection is Live!\n",
      "FishEnrichr ... Connection is Live!\n",
      "OxEnrichr ... Connection is Live!\n",
      "\n",
      "Attaching package: generics\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,\n",
      "    setequal, union\n",
      "\n",
      "\n",
      "Attaching package: BiocGenerics\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    anyDuplicated, aperm, append, as.data.frame, basename, cbind,\n",
      "    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,\n",
      "    get, grep, grepl, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, saveRDS, table, tapply, unique,\n",
      "    unsplit, which.max, which.min\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Attaching package: S4Vectors\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    rename\n",
      "\n",
      "The following object is masked from package:utils:\n",
      "\n",
      "    findMatches\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    expand.grid, I, unname\n",
      "\n",
      "\n",
      "Attaching package: IRanges\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    slice\n",
      "\n",
      "\n",
      "Attaching package: AnnotationDbi\n",
      "\n",
      "The following object is masked from package:clusterProfiler:\n",
      "\n",
      "    select\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: dplyr\n",
      "\n",
      "The following object is masked from package:AnnotationDbi:\n",
      "\n",
      "    select\n",
      "\n",
      "The following objects are masked from package:IRanges:\n",
      "\n",
      "    collapse, desc, intersect, setdiff, slice, union\n",
      "\n",
      "The following objects are masked from package:S4Vectors:\n",
      "\n",
      "    first, intersect, rename, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:Biobase:\n",
      "\n",
      "    combine\n",
      "\n",
      "The following objects are masked from package:BiocGenerics:\n",
      "\n",
      "    combine, intersect, setdiff, setequal, union\n",
      "\n",
      "The following object is masked from package:generics:\n",
      "\n",
      "    explain\n",
      "\n",
      "The following objects are masked from package:stats:\n",
      "\n",
      "    filter, lag\n",
      "\n",
      "The following objects are masked from package:base:\n",
      "\n",
      "    intersect, setdiff, setequal, union\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Reading gene list from: data/negKG_louvain_communities/24/motif_GCCTCC_genes.txt\"\n",
      "[1] \"Performing GO enrichment analysis for ontology: CC\"\n",
      "[1] \"Generating plots...\"\n",
      "[1] \"Saving results to directory: data/negKG_louvain_communities/24/enrich_GCCTCC\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.\n",
      " Please use `linewidth` instead.\n",
      " The deprecated feature was likely used in the ggtangle package.\n",
      "  Please report the issue to the authors. \n",
      "2: In fortify(object, showCategory = showCategory, by = x, ...) :\n",
      "  Arguments in `...` must be used.\n",
      " Problematic argument:\n",
      " by = x\n",
      " Did you misspell an argument name?\n",
      "3: `aes_string()` was deprecated in ggplot2 3.0.0.\n",
      " Please use tidy evaluation idioms with `aes()`.\n",
      " See also `vignette(\"ggplot2-in-packages\")` for more information.\n",
      " The deprecated feature was likely used in the enrichplot package.\n",
      "  Please report the issue at\n",
      "  <https://github.com/GuangchuangYu/enrichplot/issues>. \n"
     ]
    }
   ],
   "source": [
    "# cellular component enrichment as well\n",
    "\n",
    "neg_gene_sets = [f for f in os.listdir(\"data/neg_gene_sets\") if f.endswith(\"_genes.txt\")]\n",
    "for gs in neg_gene_sets:\n",
    "    os.system(f\"Rscript enrich.R --gene_list data/neg_gene_sets/{gs} --outdir data/neg_gene_sets/enrich_{gs.split('_')[1]} --ont CC\")\n",
    "\n",
    "for c in os.listdir(\"data/negKG_louvain_communities\"):\n",
    "    neg_gene_sets = [f for f in os.listdir(f\"data/negKG_louvain_communities/{c}\") if f.endswith(\"_genes.txt\")]\n",
    "    for gs in neg_gene_sets:\n",
    "        os.system(f\"Rscript enrich.R --gene_list data/negKG_louvain_communities/{c}/{gs} --outdir data/negKG_louvain_communities/{c}/enrich_{gs.split('_')[1]} --ont CC\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "hsg",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
